--- kdesdk-3.0.3/kbabel/catalogmanager/catalogmanager.cpp--	2002-08-26 14:38:17.000000000 +0200
+++ kdesdk-3.0.3/kbabel/catalogmanager/catalogmanager.cpp	2002-08-26 14:43:51.000000000 +0200
@@ -286,6 +286,7 @@ void CatalogManager::saveSettings()
     config->writeEntry("FileCommandNames",_settings.fileCommandNames);
 
     config->sync();
+    delete config;
 }
 
 void CatalogManager::restoreSettings()
@@ -340,6 +341,7 @@ void CatalogManager::restoreSettings()
     }
 
     KConfigGroupSaver misc(config,"Misc");
+    delete config;
 }
 
 void CatalogManager::setPreferredWindow(WId window)
@@ -527,6 +529,7 @@ void CatalogManager::find()
 	_foundFilesList.clear();
 	kdDebug() << "Calling catalogmanagerview::find" << endl;
 	QString url = _catalogManager->find(_findOptions, _toBeSearched );
+	delete config;
 	if( _catalogManager->isStopped() ) return;
 	if( !url.isEmpty() )
 	{
@@ -607,6 +610,7 @@ void CatalogManager::replace()
 
 	_foundFilesList.clear();
 	QString url = _catalogManager->find(options, _toBeSearched );
+	delete config;
 	if( _catalogManager->isStopped() ) return;
 	if( !url.isEmpty() )
 	{
