diff -urN -x CVS kdesdk.orig/debian/catalogmanager.1 kdesdk/debian/catalogmanager.1
--- kdesdk.orig/debian/catalogmanager.1	Sat Jan 31 11:41:56 2004
+++ kdesdk/debian/catalogmanager.1	Sat Oct 16 02:05:48 2004
@@ -2,7 +2,7 @@
 .\" First parameter, NAME, should be all caps
 .\" Second parameter, SECTION, should be 1-8, maybe w/ subsection
 .\" other parameters are allowed: see man(7), man(1)
-.TH CATALOGMANAGER 1 "January 31, 2004"
+.TH CATALOGMANAGER 1 "October 15, 2004"
 .\" Please adjust this date whenever revising the manpage.
 .\"
 .\" Some roff macros, for reference:
@@ -56,8 +56,9 @@
 .SH AUTHOR
 KBabel was written by Matthias Kiefer <kiefer@kde.org>,
 Wolfram Diestel <wolfram@steloj.de>, Andrea Rizzi <rizzi@kde.org>,
-Stanislav Visnovsky <visnovsky@kde.org>, Marco Wegner <dubbleu@web.de>
-and Dwayne Bailey <dwayne@translate.org.za>.
+Stanislav Visnovsky <visnovsky@kde.org>, Marco Wegner <dubbleu@web.de>,
+Dwayne Bailey <dwayne@translate.org.za>,
+Bram Schoenmakers <bramschoenmakers@kde.nl> and others.
 .br
 This manual page was prepared by Ben Burton <bab@debian.org>
 for the Debian GNU/Linux system (but may be used by others).
diff -urN -x CVS kdesdk.orig/debian/cervisia.override kdesdk/debian/cervisia.override
--- kdesdk.orig/debian/cervisia.override	Sat Jan 31 11:41:56 2004
+++ kdesdk/debian/cervisia.override	Fri Oct 15 08:46:17 2004
@@ -1,5 +1,3 @@
-# There is no -dev package.
-cervisia: non-dev-pkg-with-shlib-symlink usr/lib/libcvsservice.so.0.0.1 usr/lib/libcvsservice.so
 # Menu items
 cervisia: menu-item-contains-unknown-tag kderemove /usr/lib/menu/cervisia:9
 # kconf_update scripts don't need to be executable.
diff -urN -x CVS kdesdk.orig/debian/cervisia.xpm kdesdk/debian/cervisia.xpm
--- kdesdk.orig/debian/cervisia.xpm	Sat Jan 31 11:41:56 2004
+++ kdesdk/debian/cervisia.xpm	Fri Oct 15 16:35:50 2004
@@ -1,96 +1,195 @@
 /* XPM */
-static char *cervisia[] = {
-/* columns rows colors chars-per-pixel */
-"32 32 58 1",
-"  c #000000",
-". c #0B0000",
-"X c #140000",
-"o c #190000",
-"O c #240000",
-"+ c #2C0000",
-"@ c #320000",
-"# c #3A0000",
-"$ c #420000",
-"% c #490000",
-"& c #560202",
-"* c #5D0202",
-"= c #5C0A0A",
-"- c #630202",
-"; c #610909",
-": c #6B0000",
-"> c #730202",
-", c #7B0101",
-"< c #830000",
-"1 c #860808",
-"2 c #8B0202",
-"3 c #8A0909",
-"4 c #821F1F",
-"5 c #930101",
-"6 c #9C0000",
-"7 c #9E0F0F",
-"8 c #A30000",
-"9 c #A60B0B",
-"0 c #AB0000",
-"q c #AC0B0B",
-"w c #AF1414",
-"e c #B20101",
-"r c #B20B0B",
-"t c #B90B0B",
-"y c #B01010",
-"u c #B91010",
-"i c #A62E2E",
-"p c #A23434",
-"a c #B52828",
-"s c #B33E3E",
-"d c #C52F2F",
-"f c #CA2E2E",
-"g c #C23232",
-"h c #C53D3D",
-"j c #C83131",
-"k c #CC3B3B",
-"l c #D13838",
-"z c #D93D3D",
-"x c #C54545",
-"c c #CF4040",
-"v c #D34343",
-"b c #DC4141",
-"n c #DF5757",
-"m c #E04545",
-"M c #FE8A8A",
-"N c #FF9393",
-"B c gray100",
-"V c None",
-/* pixels */
-"VV.@++@@@@+@@O VVX@++@+@@@@@OVVV",
-"VXszrryyyrryy6% Oxkrrryyyqyq6# V",
-"V&Mkettet0q895> ;Mjeteeteq092>XV",
-"V;zteeee008862>X=zteeee008662>oV",
-"V*zreee0008862>.=bteeee000862>oV",
-"V;br0ee0008862:.=br000008866<:oV",
-"V*bq800088666<:.=bq8000888651:oV",
-"V;zq888886665,:X;z0888886655,-oV",
-"V#r6555522<<<>- $r6555222<<,>*. ",
-"V %>>>>>::-**&O V%:::>>:---*&O  ",
-"VV .......@@@@++++@#+..o@@@@@o V",
-"VVVVVVVV.pkqqwyqyqwq5#V4kqqwq<XV",
-"VVVVVVVV&Mkettter0q95>ohnerr92@ ",
-"VVVVVVVV;mteeee008863>+afee063# ",
-"VVVVVVVV*breee0e08862:Oafe0063# ",
-"VVVVVVVV-br00e000866<:Oaf00063@ ",
-"VVVVVVVV*br808008865<:Oaf0085,@ ",
-"VVVVVVVV;zq888886665,:Oad8865,@ ",
-"VVVVVVVV$r65555322<<>;o7w525<>+ ",
-"VVVVVVVVV%>>:>>::--**O $,::-:*. ",
-"VVV$++++++@##+XoXX@#++++####+.  ",
-"VXilrrrrrr9w95@ .pg9qqrww9992@ V",
-"V&Mvettreqqq05> =Nketrteqqq85:.V",
-"V;mreeee008882>X=bteeee008863>oV",
-"V*zteeee000862>.=breeee008862>oV",
-"V*br0e0008886<>.=be00e000866<:oV",
-"V*br000008866<:.=br000008866<:oV",
-"V;bq808888655<:.=b0888888655<-oV",
-"V$u6555555222>: %t65555552<<>-. ",
-"V.&,,,>,>::::-@ .*<,,,>>>::--+  ",
-"VV oooooooooo.  V oooooooooo.  V",
-"VVVV         VVVVVV          VVV"
-};
+static char * cervisia_xpm[] = {
+"32 32 160 2",
+"  	c None",
+". 	c #210000",
+"+ 	c #220101",
+"@ 	c #2A0000",
+"# 	c #2D0001",
+"$ 	c #2E0102",
+"% 	c #310002",
+"& 	c #330000",
+"* 	c #340100",
+"= 	c #370001",
+"- 	c #380002",
+"; 	c #3A0000",
+"> 	c #3B0000",
+", 	c #3C0100",
+"' 	c #3C0100",
+") 	c #410003",
+"! 	c #4B0000",
+"~ 	c #4C0000",
+"{ 	c #560000",
+"] 	c #570001",
+"^ 	c #5A0200",
+"/ 	c #5E0000",
+"( 	c #5F0000",
+"_ 	c #600001",
+": 	c #5B0300",
+"< 	c #610002",
+"[ 	c #620003",
+"} 	c #5C0400",
+"| 	c #640100",
+"1 	c #680000",
+"2 	c #630205",
+"3 	c #690000",
+"4 	c #6A0001",
+"5 	c #650300",
+"6 	c #6B0002",
+"7 	c #6C0000",
+"8 	c #6C0004",
+"9 	c #660400",
+"0 	c #6D0100",
+"a 	c #730000",
+"b 	c #6E0300",
+"c 	c #740001",
+"d 	c #750000",
+"e 	c #750002",
+"f 	c #770000",
+"g 	c #760004",
+"h 	c #700400",
+"i 	c #770105",
+"j 	c #790200",
+"k 	c #710601",
+"l 	c #780206",
+"m 	c #7E0001",
+"n 	c #7F0002",
+"o 	c #800000",
+"p 	c #800003",
+"q 	c #7A0400",
+"r 	c #820000",
+"s 	c #810004",
+"t 	c #730903",
+"u 	c #7B0600",
+"v 	c #830200",
+"w 	c #880001",
+"x 	c #830206",
+"y 	c #890002",
+"z 	c #840300",
+"A 	c #8C0000",
+"B 	c #8B0003",
+"C 	c #8C0004",
+"D 	c #850500",
+"E 	c #8D0100",
+"F 	c #8D0105",
+"G 	c #930000",
+"H 	c #8E0300",
+"I 	c #940001",
+"J 	c #950002",
+"K 	c #970000",
+"L 	c #960003",
+"M 	c #980000",
+"N 	c #860809",
+"O 	c #980004",
+"P 	c #900500",
+"Q 	c #990200",
+"R 	c #9D0000",
+"S 	c #910700",
+"T 	c #9F0000",
+"U 	c #A00001",
+"V 	c #A10002",
+"W 	c #A20000",
+"X 	c #9A0500",
+"Y 	c #8A0D04",
+"Z 	c #A40100",
+"` 	c #9B0700",
+" .	c #A90000",
+"..	c #AB0000",
+"+.	c #AC0001",
+"@.	c #A50400",
+"#.	c #AD0001",
+"$.	c #A50405",
+"%.	c #AE0002",
+"&.	c #AC0008",
+"*.	c #AF0000",
+"=.	c #AD0009",
+"-.	c #AE010A",
+";.	c #B40000",
+">.	c #B50000",
+",.	c #B00300",
+"'.	c #A60706",
+").	c #B00303",
+"!.	c #B70007",
+"~.	c #B80008",
+"{.	c #A70907",
+"].	c #B90009",
+"^.	c #B10604",
+"/.	c #A70A0E",
+"(.	c #B1060C",
+"_.	c #A80C07",
+":.	c #9F0F11",
+"<.	c #B20905",
+"[.	c #AA0E08",
+"}.	c #B2090D",
+"|.	c #BC050B",
+"1.	c #B30C06",
+"2.	c #BB0613",
+"3.	c #AB1109",
+"4.	c #B30C0E",
+"5.	c #AE1714",
+"6.	c #B72327",
+"7.	c #B92528",
+"8.	c #A62F2C",
+"9.	c #A13231",
+"0.	c #A63634",
+"a.	c #C22F2E",
+"b.	c #C82C30",
+"c.	c #CA2D2C",
+"d.	c #CA2D31",
+"e.	c #C4312F",
+"f.	c #CB2E2D",
+"g.	c #B43C3C",
+"h.	c #CC3935",
+"i.	c #D23637",
+"j.	c #C53C3E",
+"k.	c #CC393B",
+"l.	c #CD3A3C",
+"m.	c #CE3C3D",
+"n.	c #D13E3F",
+"o.	c #C64547",
+"p.	c #DA3D3D",
+"q.	c #DB3F3E",
+"r.	c #D64242",
+"s.	c #DC403F",
+"t.	c #D04646",
+"u.	c #DD4140",
+"v.	c #DF4240",
+"w.	c #E14542",
+"x.	c #E05659",
+"y.	c #FF8788",
+"z.	c #FF8889",
+"A.	c #FF898A",
+"B.	c #FF8D8B",
+"C.	c #FE9494",
+"                                                                ",
+"    g.r.}.4.4.4.4.}.}.}.}.V !     o.m.}.4.4.}.4.}.}.}.}.Q )     ",
+"    y.h.;.!.1.1.<.^.-.{.V P g     y.e.1.!.1.1.<.%.-.{.$.P a     ",
+"    u.1.;.;.>.>.%.+.@.V ` F g     w.].>.;.>.>.%.+.@.W V B k     ",
+"    r.!.,.>.*.,.+. .@.V V D k     u.!.,.>.+.,.%. .@.V ` B h     ",
+"    v.4.+.*.*.%.+.@. .V Q B 8     s.%.+.*.*.%.@.@.@.Q V x 8     ",
+"    s.3.@. .+.@. .V @.R Q w 8     r.3. . .+.+. .Z @.Q Q x 6     ",
+"    q.-.@.@.@.@.W V Q L J r 7     s.+.@.@.@.@.W V R L L r 9     ",
+"    <.V G G L P E E E r p k 9     1.L J J G P B w E r p g [     ",
+"    ! h g a h a b 6 / / / ] +     ! h 6 6 h 8 b | / / : { .     ",
+"                                                                ",
+"                  0.m.-./.5.}.}.}.}._._.L ,     k.{.{.5.}.s     ",
+"                  y.m.1.!.1.1.<._.}._.'.L a   j.x.!.^.}.'.B     ",
+"                  w.1.>.>.>.!.%.%. .V ` E h   7.f.*.>.@.V Y     ",
+"                  u.!.*.>.*.%.*. . .@.Q B h   6.c.+.>.+./.N     ",
+"                  r.4.+.+.*.,.+.@.V V V D 6   6.e.*. .@.L D     ",
+"                  s.3. .@.+. . .@.V Q L z 6   7.a. . .@.L u     ",
+"                  q.-.@. .@.@.V V Q L L p 9   7.b.@.@.@.L l     ",
+"                  <.Q P J S J E E E z r g 2   S 5.G P F x i     ",
+"                  ! g g h a h a a 6 | < ] @     p 8 6 9 6 }     ",
+"                                                                ",
+"    8.i.}.}.}.}.}.}._./.-.P >     9.a.{._.}.}.}._././.'.E       ",
+"    B.n.1.1.1.1.}.4.-._.&.L g     C.k.].1.1.<.4.^.-._.'.L h     ",
+"    w.1.>.>.>.!.*.+.@.V Q F g     w.1.;.>.>.%.%.+.@.V V D g     ",
+"    r.].;.>.*.*.+.+. .V ` D g     u.1.%.>.>.>.+.+.@.V V D k     ",
+"    v.4.+.*.*.%.*. .$.V V E h     s.<.,.*. .+.+.@.@.` Q B 8     ",
+"    s.}. . .@.+. .@.W Q L z 8     r.%. .+. . . . .W V L w 8     ",
+"    s._.@.@. .@.@.W Q Q Q s 7     u.-.@.@.@.@.@.V R Q Q p 9     ",
+"    |.V Q Q L L J J E w z t 9     4.V Q Q L L P G w z z l 2     ",
+"    ] q q q j j a b a 7 6 9 %     < u q q q j a h b 6 9 9 $     ",
+"                                                                ",
+"                                                                "};
diff -urN -x CVS kdesdk.orig/debian/changelog kdesdk/debian/changelog
--- kdesdk.orig/debian/changelog	Sun Aug  1 16:36:47 2004
+++ kdesdk/debian/changelog	Sat Oct 16 01:04:55 2004
@@ -1,11 +1,19 @@
-kdesdk (4:3.2.92-1) experimental; urgency=low
+kdesdk (4:3.3.0-1) unstable; urgency=low
 
   * New upstream release.
-  * Binary package kmtrace has returned.
+  * New package kcachegrind-converters.  These foreign format converters
+    are packaged separately so that kcachegrind need not depend on php.
+  * Binary package kmtrace has returned.  Note that the kmtrace libraries
+    for debian are now installed in /usr/lib/kmtrace, not in /usr/lib as
+    is done upstream.
   * Versioned libqt3-mt-dev build-depends is no longer necessary since
     kdelibs4-dev already forces Qt 3.3.
+  * Made libcvsservice-dev depend on kdelibs4-dev.
+  * Removed python dependency from poxml, since all apps are now compiled
+    binaries.
+  * Added a sample kde-buildrc to the kdesdk-scripts documentation.
 
- -- Ben Burton <bab@debian.org>  Sun,  1 Aug 2004 18:41:10 +1000
+ -- Ben Burton <bab@debian.org>  Fri, 15 Oct 2004 09:55:26 +1000
 
 kdesdk (4:3.2.3-2) unstable; urgency=low
 
diff -urN -x CVS kdesdk.orig/debian/control kdesdk/debian/control
--- kdesdk.orig/debian/control	Sun Aug 22 05:05:05 2004
+++ kdesdk/debian/control	Sat Oct 16 03:42:55 2004
@@ -2,14 +2,14 @@
 Section: kde
 Priority: optional
 Maintainer: Ben Burton <bab@debian.org>
-Build-Depends: automake1.7, binutils-dev, bison, debhelper (>> 4.0.0), flex, kdelibs4-dev (>= 4:3.2.92), libdb4.0-dev, libkcal2-dev (>= 4:3.2.92)
+Build-Depends: automake1.7, binutils-dev, bison, debhelper (>> 4.0.0), flex, kdelibs4-dev (>= 4:3.3.0), libdb4.2-dev, libkcal2-dev (>= 4:3.3.0)
 Standards-Version: 3.6.1
 
 Package: kdesdk
 Architecture: any
 Section: kde
 Depends: cervisia (>= ${Source-Version}), kapptemplate (>= ${Source-Version}), kbabel (>= ${Source-Version}), kbugbuster (>= ${Source-Version}), kcachegrind (>= ${Source-Version}), kdesdk-kfile-plugins (>= ${Source-Version}), kdesdk-misc (>= ${Source-Version}), kdesdk-scripts (>= ${Source-Version}), kmtrace (>= ${Source-Version}), kompare (>= ${Source-Version}), kspy (>= ${Source-Version}), kuiviewer (>= ${Source-Version}), poxml (>= ${Source-Version}), umbrello (>= ${Source-Version})
-Suggests: kbabel-dev, kdesdk-doc-html
+Suggests: kcachegrind-converters, kbabel-dev, kdesdk-doc-html
 Description: KDE Software Development Kit
  The KDE Software Development Kit is a collection of applications and tools
  useful for developing KDE applications.
@@ -115,7 +115,7 @@
 Architecture: any
 Section: devel
 Depends: ${shlibs:Depends}
-Recommends: valgrind-calltree
+Recommends: valgrind-callgrind | valgrind-calltree
 Suggests: kcachegrind-converters, graphviz, khelpcenter, kdesdk-doc-html
 Description: visualisation tool for valgrind profiling output
  KCachegrind is a visualisation tool for the profiling data generated
@@ -128,19 +128,27 @@
 Package: kcachegrind-converters
 Architecture: all
 Section: devel
-Depends: ${perl:Depends}, php4-cgi, python
+Depends: ${perl:Depends}, php4-cli, python
 Recommends: kcachegrind
-Description: TODO
- TODO
+Description: format converters for KCachegrind profiling visualisation tool
+ This is a collection of scripts for converting the output from
+ different profiling tools into a format that KCachegrind can use.
+ .
+ KCachegrind is a visualisation tool for the profiling data generated
+ by calltree, a profiling skin for valgrind.  Applications can be
+ profiled using calltree without being recompiled, and shared libraries
+ and plugin architectures are supported.
+ .
+ This package is part of the KDE Software Development Kit.
 
 Package: kdesdk-kfile-plugins
 Architecture: any
 Section: kde
 Depends: ${shlibs:Depends}
-Description: KDE file dialog plugins for source and patch files
+Description: KDE file dialog plugins for software development files
  This is a collection of plugins for the KDE file dialog.  These plugins
- extend the file dialog to offer advanced meta-information for source
- and patch files.
+ extend the file dialog to offer advanced meta-information for source files,
+ patch files and Qt Linguist data.
  .
  This package is part of the KDE Software Development Kit.
 
@@ -188,8 +196,7 @@
 Package: kmtrace
 Architecture: any
 Section: devel
-Depends: ${shlibs:Depends}
-Recommends: less
+Depends: ${shlibs:Depends}, less
 Description: a KDE memory leak tracer
  KMtrace is a KDE tool to assist with malloc debugging using glibc's
  "mtrace" functionality.
@@ -202,8 +209,9 @@
 Depends: ${shlibs:Depends}
 Suggests: khelpcenter
 Description: a KDE GUI for viewing differences between files
- Kompare is a KDE GUI for viewing the differences between two files.
- It will also optionally create a diff file.
+ Kompare is a KDE GUI for viewing the differences between files.
+ It can compare two documents, create a diff file, display a diff file
+ and/or blend a diff file back into the original documents.
  .
  This package is part of the KDE Software Development Kit.
 
@@ -258,7 +266,7 @@
 Package: libcvsservice-dev
 Architecture: any
 Section: libdevel
-Depends: libcvsservice0 (= ${Source-Version})
+Depends: libcvsservice0 (= ${Source-Version}), kdelibs4-dev
 Replaces: cervisia (<< 4:3.2.0)
 Description: development files for CVS DCOP service
  The library libcvsservice provides a DCOP service for accessing and
@@ -274,7 +282,7 @@
 Package: poxml
 Architecture: any
 Section: devel
-Depends: ${shlibs:Depends}, python
+Depends: ${shlibs:Depends}
 Description: tools for using PO-files to translate DocBook XML files
  This is a collection of tools that facilitate translating DocBook XML
  files using gettext message files (PO-files).
diff -urN -x CVS kdesdk.orig/debian/copyright kdesdk/debian/copyright
--- kdesdk.orig/debian/copyright	Sun Feb  1 08:06:59 2004
+++ kdesdk/debian/copyright	Sat Oct 16 03:42:55 2004
@@ -11,14 +11,15 @@
 
   Note that kdesdk has many contributers beyond those listed below.
 
-  Cervisia: Bernd Gehrmann <bernd@physik.hu-berlin.de>,
-      Christian Loose <christian.loose@hamburg.de> and
-      Andre Woebbeking <woebbeking@web.de>
+  Cervisia: Bernd Gehrmann <bernd@mail.berlios.de>,
+      Christian Loose <christian.loose@hamburg.de>,
+      Andre Woebbeking <woebbeking@web.de> and others
   KAppTemplate: Kurt Granroth <granroth@kde.org>
   KBabel: Matthias Kiefer <kiefer@kde.org>,
       Wolfram Diestel <wolfram@steloj.de>, Andrea Rizzi <rizzi@kde.org>,
-      Stanislav Visnovsky <visnovsky@kde.org>, Marco Wegner <dubbleu@web.de>
-      and Dwayne Bailey <dwayne@translate.org.za>
+      Stanislav Visnovsky <visnovsky@kde.org>, Marco Wegner <dubbleu@web.de>,
+      Dwayne Bailey <dwayne@translate.org.za>,
+      Bram Schoenmakers <bramschoenmakers@kde.nl> and others
   KBugBuster: Martijn Klingens <klingens@kde.org>,
       Simon Hausmann <hausmann@kde.org>,
       Cornelius Schumacher <schumacher@kde.org>, David Faure <faure@kde.org>
@@ -27,16 +28,17 @@
   KDEPalettes: Daniel M. Duley <mosfet@jorsm.com>
   KMTrace: Waldo Bastian <bastian@kde.org>, Mike Haertel <mike@ai.mit.edu>,
       Andi Kleen <ak@suse.de>
-  Kompare: John Firebaugh <jfirebaugh@kde.org> and
-      Otto Bruggeman <otto.bruggeman@home.nl>
+  Kompare: John Firebaugh <jfirebaugh@kde.org>,
+      Otto Bruggeman <otto.bruggeman@home.nl> and others
   KProfileMethod: David Faure <faure@kde.org>
   KSpy: Richard Moore <rich@kde.org>
-  KStartPerf: Geert Jansen <jansen@kde.org>
+  KStartPerf: Geert Jansen <jansen@kde.org>, Rik Hemsley <rik@kde.org>,
+      Simon Hausmann <hausmann@kde.org> and Bill Soudan <soudan@kde.org>
   KUIViewer: Richard Moore <rich@kde.org> and
       Ian Reinhart Geiser <geiseri@kde.org>
   PO-XML: Stephan Kulow <coolo@kde.org>
   Style guide check: Maksim Orlovich <orlovich@cs.rochester.edu>,
-      Ryan Cumming <bodnar42@phalynx.dhs.org> and others
+      Ryan Cumming <ryan@completely.kicks-ass.org> and others
   Umbrello: Paul Hensgen <phensgen@sourceforge.net> and several others
 
 Copyright:
@@ -44,13 +46,20 @@
   All components of this package are released under the GNU General
   Public License (GPL), with the following exceptions:
 
+    ANTLR (part of PO-XML): (see below)
     Cervisia: Q Public License (see below)
+    KCachegrind converters:
+      dprof2calltree, pprof2calltree: (see below)
     KMTrace: GNU Library General Public License (LGPL)
     KProfileMethod: GNU Library General Public License (LGPL)
     KStartPerf: Artistic License
     KUIViewer: GNU Library General Public License (LGPL)
-    Script cvs-clean: BSD License
-    Script extend_dmalloc: (see below)
+    Scripts:
+      cheatmake, kde-spellcheck2.pl: GNU Lesser General Public License (LGPL)
+      cvs-clean, cvslastreferenced: BSD License
+      cvscheck, kde-spellcheck.pl: Artistic License
+      extend_dmalloc: (see below)
+      pruneemptydirs, rc2kcfgxt.pl: Released to the public domain
 
   On Debian GNU/Linux systems, the complete texts of these licenses
   can be found in `/usr/share/common-licenses/GPL',
@@ -59,6 +68,30 @@
 
   --------------------
 
+  ANTLR:
+
+  SOFTWARE RIGHTS
+
+  ANTLR 2.6.0 MageLang Insitute, 1998, 1999
+
+  We reserve no legal rights to the ANTLR--it is fully in the public
+  domain. An individual or company may do whatever they wish with source
+  code distributed with ANTLR or the code generated by ANTLR, including
+  the incorporation of ANTLR, or its output, into commercial software.
+
+  We encourage users to develop software with ANTLR. However, we do ask
+  that credit is given to us for developing ANTLR. By "credit", we mean
+  that if you use ANTLR or incorporate any source code into one of your
+  programs (commercial product, research project, or otherwise) that you
+  acknowledge this fact somewhere in the documentation, research report,
+  etc... If you like ANTLR and have developed a nice tool with the
+  output, please mention that you developed it using ANTLR. In addition,
+  we ask that the headers remain intact in our source code. As long as
+  these guidelines are kept, we expect to continue enhancing this system
+  and expect to make other tools available as they are completed.
+
+  --------------------
+
   Cervisia:
 
                                THE Q PUBLIC LICENSE
@@ -167,6 +200,44 @@
 
   --------------------
 
+  Converters dprof2calltree, pprof2calltree for KCachegrind:
+
+  Redistribution and use in source and binary forms, with or without
+  modification, are permitted provided that the following conditions are met:
+
+  - Redistributions of source code must retain the above copyright notice,
+  this list of conditions and the following disclaimer.
+
+  - Redistributions in binary form must reproduce the above copyright
+  notice, this list of conditions and the following disclaimer in the
+  documentation and/or other materials provided with the distribution.
+
+  - All advertising materials mentioning features or use of this software
+  must display the following acknowledgement: This product includes software
+  developed by OmniTI Computer Consulting.
+
+  - Neither name of the company nor the names of its contributors may be
+  used to endorse or promote products derived from this software without
+  specific prior written permission.
+
+  THIS SOFTWARE IS PROVIDED BY THE REGENTS AND CONTRIBUTORS `AS IS'' AND ANY
+  EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+  WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+  DISCLAIMED. IN NO EVENT SHALL THE REGENTS OR CONTRIBUTORS BE LIABLE FOR ANY
+  DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+  (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+  ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+  THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+  Copyright (c) 2004 OmniTI Computer Consulting
+  All rights reserved
+  The following code was written by George Schlossnagle <george@omniti.com>
+  and is provided completely free and without any warranty.
+
+  --------------------
+
   Script extend_dmalloc:
 
   Copyright 1995 by Gray Watson
diff -urN -x CVS kdesdk.orig/debian/cvsbackport.1 kdesdk/debian/cvsbackport.1
--- kdesdk.orig/debian/cvsbackport.1	Thu Jan  1 01:00:00 1970
+++ kdesdk/debian/cvsbackport.1	Fri Oct 15 16:35:50 2004
@@ -0,0 +1,51 @@
+.\"                                      Hey, EMACS: -*- nroff -*-
+.\" First parameter, NAME, should be all caps
+.\" Second parameter, SECTION, should be 1-8, maybe w/ subsection
+.\" other parameters are allowed: see man(7), man(1)
+.TH CVSBACKPORT 1 "October 15, 2004"
+.\" Please adjust this date whenever revising the manpage.
+.\"
+.\" Some roff macros, for reference:
+.\" .nh        disable hyphenation
+.\" .hy        enable hyphenation
+.\" .ad l      left justify
+.\" .ad b      justify to both left and right margins
+.\" .nf        disable filling
+.\" .fi        enable filling
+.\" .br        insert line break
+.\" .sp <n>    insert n+1 empty lines
+.\" for manpage-specific macros, see man(7)
+.SH NAME
+cvsbackport \- backport the last change in HEAD to a branch
+.SH SYNOPSIS
+.B cvsbackport
+\fIfile\fP ...
+.SH DESCRIPTION
+For each file given on the command line,
+\fBcvsbackport\fP will attempt to determine the last change made to
+the file in HEAD and then backport that change to a CVS branch (the
+branch name is hard-coded; see below).
+.PP
+The changes to be backported will be displayed on screen, and you will
+have the chance to confirm or abort before any commits are made.
+.PP
+The files must be checked out from HEAD when you run this utility.
+.PP
+This utility is part of the KDE Software Development Kit.
+.SH WARNINGS
+The branch tag is hard-coded into the script (currently it is set to
+KDE_3_3_BRANCH).
+.PP
+If you abort (i.e., you do not make the suggested commits), your files
+will be checked out from the branch (and not from HEAD as they were
+before).
+.SH SEE ALSO
+.BR cvsblame (1),
+.BR cvslastchange (1),
+.BR cvslastlog (1),
+.BR cvsrevertlast (1).
+.SH AUTHOR
+This utility was written by Dirk Mueller and David Faure.
+.br
+This manual page was prepared by Ben Burton <bab@debian.org>
+for the Debian GNU/Linux system (but may be used by others).
diff -urN -x CVS kdesdk.orig/debian/cvslastchange.1 kdesdk/debian/cvslastchange.1
--- kdesdk.orig/debian/cvslastchange.1	Sun Feb  1 08:06:59 2004
+++ kdesdk/debian/cvslastchange.1	Fri Oct 15 16:35:50 2004
@@ -2,7 +2,7 @@
 .\" First parameter, NAME, should be all caps
 .\" Second parameter, SECTION, should be 1-8, maybe w/ subsection
 .\" other parameters are allowed: see man(7), man(1)
-.TH CVSLASTCHANGE 1 "January 31, 2004"
+.TH CVSLASTCHANGE 1 "October 15, 2004"
 .\" Please adjust this date whenever revising the manpage.
 .\"
 .\" Some roff macros, for reference:
@@ -40,6 +40,7 @@
 \fBM\fP +\fIdigits\fP -\fIdigits\fP
 These options are currently ignored.
 .SH SEE ALSO
+.BR cvsbackport (1),
 .BR cvsblame (1),
 .BR cvslastlog (1),
 .BR cvsrevertlast (1).
diff -urN -x CVS kdesdk.orig/debian/cvslastlog.1 kdesdk/debian/cvslastlog.1
--- kdesdk.orig/debian/cvslastlog.1	Wed Dec 10 07:08:45 2003
+++ kdesdk/debian/cvslastlog.1	Fri Oct 15 16:35:50 2004
@@ -2,7 +2,7 @@
 .\" First parameter, NAME, should be all caps
 .\" Second parameter, SECTION, should be 1-8, maybe w/ subsection
 .\" other parameters are allowed: see man(7), man(1)
-.TH CVSLASTLOG 1 "May 12, 2003"
+.TH CVSLASTLOG 1 "October 15, 2004"
 .\" Please adjust this date whenever revising the manpage.
 .\"
 .\" Some roff macros, for reference:
@@ -27,6 +27,7 @@
 .PP
 This utility is part of the KDE Software Development Kit.
 .SH SEE ALSO
+.BR cvsbackport (1),
 .BR cvsblame (1),
 .BR cvslastchange (1),
 .BR cvsrevertlast (1).
diff -urN -x CVS kdesdk.orig/debian/cvsrevertlast.1 kdesdk/debian/cvsrevertlast.1
--- kdesdk.orig/debian/cvsrevertlast.1	Wed Dec 10 07:08:45 2003
+++ kdesdk/debian/cvsrevertlast.1	Fri Oct 15 16:35:50 2004
@@ -2,7 +2,7 @@
 .\" First parameter, NAME, should be all caps
 .\" Second parameter, SECTION, should be 1-8, maybe w/ subsection
 .\" other parameters are allowed: see man(7), man(1)
-.TH CVSREVERTLAST 1 "May 12, 2003"
+.TH CVSREVERTLAST 1 "October 15, 2004"
 .\" Please adjust this date whenever revising the manpage.
 .\"
 .\" Some roff macros, for reference:
@@ -26,6 +26,7 @@
 .PP
 This utility is part of the KDE Software Development Kit.
 .SH SEE ALSO
+.BR cvsbackport (1),
 .BR cvsblame (1),
 .BR cvslastchange (1),
 .BR cvslastlog (1).
diff -urN -x CVS kdesdk.orig/debian/cvsservice.1 kdesdk/debian/cvsservice.1
--- kdesdk.orig/debian/cvsservice.1	Mon Mar  8 00:59:39 2004
+++ kdesdk/debian/cvsservice.1	Fri Oct 15 18:48:14 2004
@@ -2,7 +2,7 @@
 .\" First parameter, NAME, should be all caps
 .\" Second parameter, SECTION, should be 1-8, maybe w/ subsection
 .\" other parameters are allowed: see man(7), man(1)
-.TH CVSSERVICE 1 "February 6, 2004"
+.TH CVSSERVICE 1 "October 15, 2004"
 .\" Please adjust this date whenever revising the manpage.
 .\"
 .\" Some roff macros, for reference:
@@ -64,13 +64,14 @@
 .BR cvsaskpass (1),
 .BR dcop (1).
 .PP
-Details of how to use the CVS service from within another script or C++
-program can be found in \fI/usr/share/doc/libcvsservice0/DESIGN\fP.
+Details of how to use the CVS service from within a shell script, a C++
+program or through the JavaScript bindings can be found in
+\fI/usr/share/doc/libcvsservice0/DESIGN\fP.
 .SH AUTHOR
-The CVS service was written by Christian Loose <christian.loose@hamburg.de>.
+The CVS service was written by Christian Loose <christian.loose@kdemail.net>.
 .br
 Cervisia was written by Bernd Gehrmann <bernd@physik.hu-berlin.de>,
-Christian Loose <christian.loose@hamburg.de> and
+Christian Loose <christian.loose@kdemail.net> and
 Andre Woebbeking <woebbeking@web.de>.
 .br
 This manual page was prepared by Ben Burton <bab@debian.org>
diff -urN -x CVS kdesdk.orig/debian/demangle.1 kdesdk/debian/demangle.1
--- kdesdk.orig/debian/demangle.1	Sun Aug  1 16:36:12 2004
+++ kdesdk/debian/demangle.1	Fri Oct 15 18:48:14 2004
@@ -50,8 +50,8 @@
 .RE
 .SH SEE ALSO
 .BR kminspector (1),
-.BR kmmatch (1).
-.BR kmtrace (1),
+.BR kmmatch (1),
+.BR kmtrace (1).
 .SH AUTHOR
 This manual page was prepared by Ben Burton <bab@debian.org>
 for the Debian GNU/Linux system (but may be used by others).
diff -urN -x CVS kdesdk.orig/debian/dprof2calltree.1 kdesdk/debian/dprof2calltree.1
--- kdesdk.orig/debian/dprof2calltree.1	Thu Jan  1 01:00:00 1970
+++ kdesdk/debian/dprof2calltree.1	Sat Oct 16 03:01:38 2004
@@ -0,0 +1,60 @@
+.\"                                      Hey, EMACS: -*- nroff -*-
+.\" First parameter, NAME, should be all caps
+.\" Second parameter, SECTION, should be 1-8, maybe w/ subsection
+.\" other parameters are allowed: see man(7), man(1)
+.TH DPROF2CALLTREE 1 "October 15, 2004"
+.\" Please adjust this date whenever revising the manpage.
+.\"
+.\" Some roff macros, for reference:
+.\" .nh        disable hyphenation
+.\" .hy        enable hyphenation
+.\" .ad l      left justify
+.\" .ad b      justify to both left and right margins
+.\" .nf        disable filling
+.\" .fi        enable filling
+.\" .br        insert line break
+.\" .sp <n>    insert n+1 empty lines
+.\" for manpage-specific macros, see man(7)
+.SH NAME
+dprof2calltree \- convert perl's DProf profiling data to KCachegrind calltree format
+.SH SYNOPSIS
+.B dprof2calltree
+\fB\-f\fP \fIinput\fP
+[ \fB\-o\fP \fIoutput\fP ]
+.SH DESCRIPTION
+\fBdprof2calltree\fP reads the \fItmon.out\fP profiling data output by
+Perl's Devel::DProf profiling package, and converts it into calltree format
+for use by KCachegrind.
+.PP
+This utility is part of the KDE Software Development Kit.
+.SH OPTIONS
+.TP
+\fB\-f\fP \fIinput\fP
+Specifies the DProf \fItmon.out\fP file to convert.
+.TP
+\fB\-o\fP \fIoutput\fP
+Write the calltree output to the given file for use by KCachegrind.
+If omitted, the output will be written to file of the form
+\fIcachgrind.out.<input>\fP.
+.SH EXAMPLE
+If you run a perl script as:
+.PP
+.RS
+perl -d:DProf yourscript.pl
+.RE
+.PP
+then this will write profiling data to \fItmon.out\fP.  You can then
+work with this data in KCachegrind as follows:
+.PP
+.RS
+dprof2calltree -f tmin.out -o cachegrind.out.foo
+.br
+kcachegrind cachegrind.out.foo
+.RE
+.SH SEE ALSO
+.BR kcachegrind (1).
+.SH AUTHOR
+This converter was written by George Schlossnagle <george@omniti.com>.
+.br
+This manual page was prepared by Ben Burton <bab@debian.org>
+for the Debian GNU/Linux system (but may be used by others).
diff -urN -x CVS kdesdk.orig/debian/extractrc.1 kdesdk/debian/extractrc.1
--- kdesdk.orig/debian/extractrc.1	Sun Feb  1 08:06:59 2004
+++ kdesdk/debian/extractrc.1	Fri Oct 15 18:48:14 2004
@@ -2,7 +2,7 @@
 .\" First parameter, NAME, should be all caps
 .\" Second parameter, SECTION, should be 1-8, maybe w/ subsection
 .\" other parameters are allowed: see man(7), man(1)
-.TH EXTRACTRC 1 "January 31, 2004"
+.TH EXTRACTRC 1 "October 15, 2004"
 .\" Please adjust this date whenever revising the manpage.
 .\"
 .\" Some roff macros, for reference:
@@ -16,28 +16,47 @@
 .\" .sp <n>    insert n+1 empty lines
 .\" for manpage-specific macros, see man(7)
 .SH NAME
-extractrc \- extract message strings from UI and GUI-RC files
+extractrc \- extract message strings from UI and XML GUI files
 .SH SYNOPSIS
 .B extractrc
+[ \fB\-\-tag\-group=\fP\fIgroup\fP ]
+[ \fB\-\-tag=\fP\fIname\fP ... ]
 [ \fB\-\-context=\fP\fIname\fP ]
-[ \fB\-\-tag=\fP\fIname\fP ]
+[ \fB\-\-lines\fP ]
 \fIfile\fP ...
+.PP
+.B extractrc \-\-help
 .SH DESCRIPTION
-\fBextractrc\fP finds all text tags and other message strings in the given
-files and writes the corresponding \fIi18n()\fP calls to standard output
-so that xgettext can parse them.
+\fBextractrc\fP finds all text tags and other message strings within the
+given files.  It then writes the corresponding \fIi18n()\fP calls to
+standard output (usually redirected to rc.cpp) so that xgettext can parse
+them.
 .PP
-It understands both (Qt/KDE)-designer's UI files and XML GUI-RC files.
+It understands both Qt designer (.ui) files and XML GUI (.rc) files.
 .PP
 This utility is part of the KDE Software Development Kit.
 .SH OPTIONS
 .TP
+\fB\-\-tag\-group=\fP\fIgroup\fP
+Use a predefined group of tags that we are interested in.
+Valid groups are default, koffice and none.
+The default group will be used if this option is omitted.
+.TP
+\fB\-\-tag=\fP\fIname\fP
+Extract also the contents of the tag \fIname\fP.
+This option may be used multiple times
+(e.g., \-\-tag=tag_one \-\-tag=tag_two).
+.TP
 \fB\-\-context=\fP\fIname\fP
 Give all i18n() calls the given context: i18n("\fIname\fP", ...).
 .TP
-\fB\-\-tag=\fP\fIname\fP
-Extract also the contents of the tag \fIname\fP (this option may be
-used multiple times).
+\fB\-\-lines\fP
+Include source line numbers in comments.
+.TP
+\fB\-\-help\fP
+Display a summary of options.
 .SH AUTHOR
+This utility was written by Richard Evans <rich@ridas.com> and others.
+.br
 This manual page was prepared by Ben Burton <bab@debian.org>
 for the Debian GNU/Linux system (but may be used by others).
diff -urN -x CVS kdesdk.orig/debian/fixuifiles.1 kdesdk/debian/fixuifiles.1
--- kdesdk.orig/debian/fixuifiles.1	Thu Jan  1 01:00:00 1970
+++ kdesdk/debian/fixuifiles.1	Fri Oct 15 16:35:50 2004
@@ -0,0 +1,59 @@
+.\"                                      Hey, EMACS: -*- nroff -*-
+.\" First parameter, NAME, should be all caps
+.\" Second parameter, SECTION, should be 1-8, maybe w/ subsection
+.\" other parameters are allowed: see man(7), man(1)
+.TH FIXUIFILES 1 "October 15, 2004"
+.\" Please adjust this date whenever revising the manpage.
+.\"
+.\" Some roff macros, for reference:
+.\" .nh        disable hyphenation
+.\" .hy        enable hyphenation
+.\" .ad l      left justify
+.\" .ad b      justify to both left and right margins
+.\" .nf        disable filling
+.\" .fi        enable filling
+.\" .br        insert line break
+.\" .sp <n>    insert n+1 empty lines
+.\" for manpage-specific macros, see man(7)
+.SH NAME
+fixuifiles \- remove undesirable features from Qt/KDE .ui files
+.SH SYNOPSIS
+.B fixuifiles
+[ \fB\-v, \-\-verbose\fP ]
+[ \fIfile\fP ... ]
+.br
+.B fixuifiles \-\-help
+.SH DESCRIPTION
+\fBfixuifiles\fP processes one or more Qt/KDE user interface definition
+files (.ui files) and tries to remove some of the insanities found
+therein.  If no filenames are given, fixuifiles will process all .ui
+files in or beneath the current directory.
+.PP
+The following problems are fixed:
+.PP
+.RS
+Too high minimum Qt version;
+.PP
+Hardcoded untranslatable Alt+Letter accels (auto-added by Qt Designer);
+.PP
+Captions that are equal to class names (auto-added by Qt Designer).
+.RE
+.PP
+The threshold Qt version can be set
+using either a \fI.qt_minversion\fP file (containing simply the Qt version
+number) or a \fIconfigure.in.in\fP file (containing an appropriate
+MIN_CONFIG macro).  These files may be in the directory containing the .ui
+file or any parent directory.  If no threshold Qt version is found,
+a sensible default will be used.
+.PP
+This utility is part of the KDE Software Development Kit.
+.SH OPTIONS
+.TP
+\fB\-v, \-\-verbose\fP
+Verbose mode.  Additional debugging information is written to standard output.
+.SH AUTHOR
+fixuifiles was written by David Faure <faure@kde.org> and
+Dirk Mueller <mueller@kde.org>.
+.br
+This manual page was prepared by Ben Burton <bab@debian.org>
+for the Debian GNU/Linux system (but may be used by others).
diff -urN -x CVS kdesdk.orig/debian/hotshot2calltree.1 kdesdk/debian/hotshot2calltree.1
--- kdesdk.orig/debian/hotshot2calltree.1	Thu Jan  1 01:00:00 1970
+++ kdesdk/debian/hotshot2calltree.1	Sat Oct 16 03:01:38 2004
@@ -0,0 +1,57 @@
+.\"                                      Hey, EMACS: -*- nroff -*-
+.\" First parameter, NAME, should be all caps
+.\" Second parameter, SECTION, should be 1-8, maybe w/ subsection
+.\" other parameters are allowed: see man(7), man(1)
+.TH HOTSHOT2CALLTREE 1 "October 15, 2004"
+.\" Please adjust this date whenever revising the manpage.
+.\"
+.\" Some roff macros, for reference:
+.\" .nh        disable hyphenation
+.\" .hy        enable hyphenation
+.\" .ad l      left justify
+.\" .ad b      justify to both left and right margins
+.\" .nf        disable filling
+.\" .fi        enable filling
+.\" .br        insert line break
+.\" .sp <n>    insert n+1 empty lines
+.\" for manpage-specific macros, see man(7)
+.SH NAME
+hotshot2calltree \- convert hotshot profiling data to KCachegrind calltree format
+.SH SYNOPSIS
+.B hotshot2calltree
+[ \fB\-\-file\-limit\fP \fIlimit\fP ]
+\fB\-o\fP \fIoutput\fP \fIinput\fP [ \fIinput\fP ... ]
+.SH DESCRIPTION
+\fBhotshot2calltree\fP reads one or more pstat profiling data files
+as output by the hotshot python profiler, and converts them into
+calltree format for use by KCachegrind.
+.PP
+This utility is part of the KDE Software Development Kit.
+.SH OPTIONS
+.TP
+\fB\-o\fP \fIoutput\fP
+Write the calltree output into the given file for use by KCachegrind.
+This may be \fB\-\fP if you wish to write to standard output.
+.TP
+\fIinput\fP ...
+Read the hotshot profiling data from the given file(s).
+.TP
+\fB\-\-file\-limit\fP \fIlimit\fP
+Stop after the given number of input files.
+.SH EXAMPLE
+Suppose that you have used hotshot to profile a python app, with results
+dumped to the file \fIpythongrind.prof\fP.  You may then work with this
+data in KCachegrind as follows.
+.PP
+.RS
+hotshot2cachegrind \-o cachegrind.out.0 pythongrind.prof
+.br
+kcachegrind cachegrind.out.0
+.RE
+.SH SEE ALSO
+.BR kcachegrind (1).
+.SH AUTHOR
+This converter was written by Jorg Beyer <job@webde-ag.de>.
+.br
+This manual page was prepared by Ben Burton <bab@debian.org>
+for the Debian GNU/Linux system (but may be used by others).
diff -urN -x CVS kdesdk.orig/debian/kbabel.1 kdesdk/debian/kbabel.1
--- kdesdk.orig/debian/kbabel.1	Sat Jan 31 11:41:56 2004
+++ kdesdk/debian/kbabel.1	Sat Oct 16 02:05:48 2004
@@ -2,7 +2,7 @@
 .\" First parameter, NAME, should be all caps
 .\" Second parameter, SECTION, should be 1-8, maybe w/ subsection
 .\" other parameters are allowed: see man(7), man(1)
-.TH KBABEL 1 "January 31, 2004"
+.TH KBABEL 1 "October 15, 2004"
 .\" Please adjust this date whenever revising the manpage.
 .\"
 .\" Some roff macros, for reference:
@@ -64,8 +64,9 @@
 .SH AUTHOR
 KBabel was written by Matthias Kiefer <kiefer@kde.org>,
 Wolfram Diestel <wolfram@steloj.de>, Andrea Rizzi <rizzi@kde.org>,
-Stanislav Visnovsky <visnovsky@kde.org>, Marco Wegner <dubbleu@web.de>
-and Dwayne Bailey <dwayne@translate.org.za>.
+Stanislav Visnovsky <visnovsky@kde.org>, Marco Wegner <dubbleu@web.de>,
+Dwayne Bailey <dwayne@translate.org.za>,
+Bram Schoenmakers <bramschoenmakers@kde.nl> and others.
 .br
 This manual page was prepared by Ben Burton <bab@debian.org>
 for the Debian GNU/Linux system (but may be used by others).
diff -urN -x CVS kdesdk.orig/debian/kbabel.shlibs kdesdk/debian/kbabel.shlibs
--- kdesdk.orig/debian/kbabel.shlibs	Sat Jan 31 11:41:56 2004
+++ kdesdk/debian/kbabel.shlibs	Fri Oct 15 08:46:17 2004
@@ -1,2 +1,2 @@
-libkbabelcommon 3 kbabel (>= 4:3.1.95-0)
-libkbabeldictplugin 1 kbabel (>= 3.0.8-0)
+libkbabelcommon 3 kbabel (>= 4:3.3.0)
+libkbabeldictplugin 1 kbabel (>= 4:3.3.0)
diff -urN -x CVS kdesdk.orig/debian/kbabeldict.1 kdesdk/debian/kbabeldict.1
--- kdesdk.orig/debian/kbabeldict.1	Sat Jan 31 11:41:56 2004
+++ kdesdk/debian/kbabeldict.1	Sat Oct 16 02:05:48 2004
@@ -2,7 +2,7 @@
 .\" First parameter, NAME, should be all caps
 .\" Second parameter, SECTION, should be 1-8, maybe w/ subsection
 .\" other parameters are allowed: see man(7), man(1)
-.TH KBABELDICT 1 "January 31, 2004"
+.TH KBABELDICT 1 "October 15, 2004"
 .\" Please adjust this date whenever revising the manpage.
 .\"
 .\" Some roff macros, for reference:
@@ -56,8 +56,9 @@
 .SH AUTHOR
 KBabel was written by Matthias Kiefer <kiefer@kde.org>,
 Wolfram Diestel <wolfram@steloj.de>, Andrea Rizzi <rizzi@kde.org>,
-Stanislav Visnovsky <visnovsky@kde.org>, Marco Wegner <dubbleu@web.de>
-and Dwayne Bailey <dwayne@translate.org.za>.
+Stanislav Visnovsky <visnovsky@kde.org>, Marco Wegner <dubbleu@web.de>,
+Dwayne Bailey <dwayne@translate.org.za>,
+Bram Schoenmakers <bramschoenmakers@kde.nl> and others.
 .br
 This manual page was prepared by Ben Burton <bab@debian.org>
 for the Debian GNU/Linux system (but may be used by others).
diff -urN -x CVS kdesdk.orig/debian/kcachegrind-converters.manpages kdesdk/debian/kcachegrind-converters.manpages
--- kdesdk.orig/debian/kcachegrind-converters.manpages	Thu Jan  1 01:00:00 1970
+++ kdesdk/debian/kcachegrind-converters.manpages	Sat Oct 16 03:01:38 2004
@@ -0,0 +1,5 @@
+debian/dprof2calltree.1
+debian/hotshot2calltree.1
+debian/memprof2calltree.1
+debian/op2calltree.1
+debian/pprof2calltree.1
diff -urN -x CVS kdesdk.orig/debian/kcachegrind.1 kdesdk/debian/kcachegrind.1
--- kdesdk.orig/debian/kcachegrind.1	Sun Feb  1 08:06:59 2004
+++ kdesdk/debian/kcachegrind.1	Sat Oct 16 03:01:38 2004
@@ -2,7 +2,7 @@
 .\" First parameter, NAME, should be all caps
 .\" Second parameter, SECTION, should be 1-8, maybe w/ subsection
 .\" other parameters are allowed: see man(7), man(1)
-.TH KCACHEGRIND 1 "January 31, 2004"
+.TH KCACHEGRIND 1 "October 15, 2004"
 .\" Please adjust this date whenever revising the manpage.
 .\"
 .\" Some roff macros, for reference:
@@ -51,6 +51,12 @@
 If the KDE Help Centre is not installed then you can install the package
 \fIkdesdk-doc-html\fP and read this documentation in HTML format from
 \fI/usr/share/doc/kde/HTML/en/kcachegrind/\fP.
+.SH SEE ALSO
+.BR dprof2calltree (1),
+.BR hotshot2calltree (1),
+.BR memprof2calltree (1),
+.BR op2calltree (1),
+.BR pprof2calltree (1).
 .SH AUTHOR
 KCachegrind was written by Josef Weidendorfer <Josef.Weidendorfer@gmx.de>.
 .br
diff -urN -x CVS kdesdk.orig/debian/kdemangen.pl.1 kdesdk/debian/kdemangen.pl.1
--- kdesdk.orig/debian/kdemangen.pl.1	Thu Jan  1 01:00:00 1970
+++ kdesdk/debian/kdemangen.pl.1	Fri Oct 15 16:35:50 2004
@@ -0,0 +1,51 @@
+.\"                                      Hey, EMACS: -*- nroff -*-
+.\" First parameter, NAME, should be all caps
+.\" Second parameter, SECTION, should be 1-8, maybe w/ subsection
+.\" other parameters are allowed: see man(7), man(1)
+.TH KDEMANGEN.PL 1 "October 15, 2004"
+.\" Please adjust this date whenever revising the manpage.
+.\"
+.\" Some roff macros, for reference:
+.\" .nh        disable hyphenation
+.\" .hy        enable hyphenation
+.\" .ad l      left justify
+.\" .ad b      justify to both left and right margins
+.\" .nf        disable filling
+.\" .fi        enable filling
+.\" .br        insert line break
+.\" .sp <n>    insert n+1 empty lines
+.\" for manpage-specific macros, see man(7)
+.SH NAME
+kdemangen.pl \- automatically create a draft manpage for a KDE app
+.SH SYNOPSIS
+.B kdemangen.pl
+.I /path/to/application
+.SH DESCRIPTION
+\fBkdemangen.pl\fP will automatically create a draft manpage for a KDE
+application.  Manpages are always better when hand-tailored, but this
+should give you a fairly good start.
+.PP
+To create the manpage, it will use the \-\-author and \-\-help\-all
+output for the application passed on the command line.
+It will also use the description in the
+Debian control file if one is available.  To make use of this additional
+feature, you must run kdemangen.pl from within the \fIdebian/\fP
+subdirectory of the application sources.
+.PP
+The new manpage will be written to standard output.
+.PP
+This utility is part of the KDE Software Development Kit.
+.SH WARNING
+This utility will only work for a full KDE application that uses
+KCmdLineArgs (which most full KDE applications do).
+.SH EXAMPLE
+.RS
+cd /path/to/kde/srcs/kdeedu/debian
+.br
+kdemangen /usr/bin/kstars > kstars.1
+.RE
+.SH AUTHOR
+kdemangen.pl was written by Dominique Devriese <devriese@kde.org>.
+.br
+This manual page was prepared by Ben Burton <bab@debian.org>
+for the Debian GNU/Linux system (but may be used by others).
diff -urN -x CVS kdesdk.orig/debian/kdesdk-doc-html.doc-base.cervisia kdesdk/debian/kdesdk-doc-html.doc-base.cervisia
--- kdesdk.orig/debian/kdesdk-doc-html.doc-base.cervisia	Sun Aug  1 10:29:31 2004
+++ kdesdk/debian/kdesdk-doc-html.doc-base.cervisia	Fri Oct 15 08:46:17 2004
@@ -1,6 +1,6 @@
 Document: kdesdk-doc-html-cervisia
 Title: Cervisia Manual
-Author: Bernd Gehrmann <bernd@mail.berlios.de>
+Author: Bernd Gehrmann <bernd@mail.berlios.de> and Carlos Woelz <carloswoelz@imap-mail.com>
 Abstract: Cervisia provides a graphical view of CVS.
 Section: Apps/Programming
 
diff -urN -x CVS kdesdk.orig/debian/kdesdk-doc-html.doc-base.kcachegrind kdesdk/debian/kdesdk-doc-html.doc-base.kcachegrind
--- kdesdk.orig/debian/kdesdk-doc-html.doc-base.kcachegrind	Sun Aug  1 10:29:31 2004
+++ kdesdk/debian/kdesdk-doc-html.doc-base.kcachegrind	Fri Oct 15 08:46:17 2004
@@ -1,7 +1,8 @@
 Document: kdesdk-doc-html-kcachegrind
 Title: KCachegrind Handbook
 Author: Josef Weidendorfer <Josef.Weidendorfer@gmx.de>
-Abstract: KCachegrind is a KDE Frontend for Cachegrind, part of Valgrind.
+Abstract: KCachegrind is a profile data visualization tool, written
+ using the KDE environment.
 Section: Apps/Programming
 
 Format: HTML
diff -urN -x CVS kdesdk.orig/debian/kdesdk-scripts.README.Debian kdesdk/debian/kdesdk-scripts.README.Debian
--- kdesdk.orig/debian/kdesdk-scripts.README.Debian	Sat Jan 31 11:41:56 2004
+++ kdesdk/debian/kdesdk-scripts.README.Debian	Fri Oct 15 08:46:17 2004
@@ -28,9 +28,12 @@
     (setq kde-full-name "Your Name")
     (setq kde-email "Your Email")
 
+  See the file /usr/share/emacs/site-lisp/kdesdk-scripts/kde-emacs.el for
+  further information regarding emacs.
+
 Shell completion controls:
 
-  Completion controls are provided in /usr/share/kdesdk-scripts/completion/
+  Completion controls are provided in /usr/share/kdesdk-scripts/completions/
   for a variety of shells.
 
   To enable completion controls for zsh, you need to add the following
@@ -59,4 +62,4 @@
 
     --suppressions=/usr/lib/valgrind/kde.supp
 
- -- Ben Burton <bab@debian.org>, Sat, 31 Jan 2004 12:08:48 +1100
+ -- Ben Burton <bab@debian.org>, Fri, 15 Oct 2004 09:55:26 +1000
diff -urN -x CVS kdesdk.orig/debian/kdesdk-scripts.docs kdesdk/debian/kdesdk-scripts.docs
--- kdesdk.orig/debian/kdesdk-scripts.docs	Thu Nov 21 05:57:28 2002
+++ kdesdk/debian/kdesdk-scripts.docs	Fri Oct 15 08:46:17 2004
@@ -1,3 +1,2 @@
 scripts/README
-scripts/kdecompile.README
 scripts/kde-emacs/kde-emacs-tips.texi
diff -urN -x CVS kdesdk.orig/debian/kdesdk-scripts.install kdesdk/debian/kdesdk-scripts.install
--- kdesdk.orig/debian/kdesdk-scripts.install	Mon Aug  2 00:38:28 2004
+++ kdesdk/debian/kdesdk-scripts.install	Fri Oct 15 08:57:45 2004
@@ -33,6 +33,7 @@
 usr/bin/pruneemptydirs
 usr/bin/qtdoc
 usr/bin/zonetab2pot.py
+usr/share/doc/kdesdk-scripts
 usr/share/man/man1/cvsblame.1
 usr/share/man/man1/cvsversion.1
 usr/share/man/man1/kde-build.1
diff -urN -x CVS kdesdk.orig/debian/kdesdk-scripts.manpages kdesdk/debian/kdesdk-scripts.manpages
--- kdesdk.orig/debian/kdesdk-scripts.manpages	Sun Aug  1 10:29:31 2004
+++ kdesdk/debian/kdesdk-scripts.manpages	Fri Oct 15 16:35:50 2004
@@ -6,6 +6,7 @@
 debian/create_makefiles.1
 debian/cvs-clean.1
 debian/cvs2dist.1
+debian/cvsbackport.1
 debian/cvscheck.1
 debian/cvslastchange.1
 debian/cvslastlog.1
@@ -14,12 +15,16 @@
 debian/extend_dmalloc.1
 debian/extractrc.1
 debian/fixkdeincludes.1
+debian/fixuifiles.1
 debian/findmissingcrystal.1
 debian/kdedoc.1
 debian/kdekillall.1
 debian/kdelnk2desktop.py.1
+debian/kdemangen.pl.1
 debian/licensecheck.1
 debian/makeobj.1
+debian/package_crystalsvg.1
+debian/png2mng.pl.1
 debian/pruneemptydirs.1
 debian/qtdoc.1
 debian/zonetab2pot.py.1
diff -urN -x CVS kdesdk.orig/debian/kminspector.1 kdesdk/debian/kminspector.1
--- kdesdk.orig/debian/kminspector.1	Sun Aug  1 16:36:12 2004
+++ kdesdk/debian/kminspector.1	Fri Oct 15 18:48:14 2004
@@ -37,8 +37,8 @@
 This utility is part of the KDE Software Development Kit.
 .SH SEE ALSO
 .BR demangle (1),
-.BR kmmatch (1).
-.BR kmtrace (1),
+.BR kmmatch (1),
+.BR kmtrace (1).
 .PP
 Much more extensive documentation for KMtrace can be found in
 \fI/usr/share/doc/kmtrace/README\fP.
diff -urN -x CVS kdesdk.orig/debian/kmtrace.README.Debian kdesdk/debian/kmtrace.README.Debian
--- kdesdk.orig/debian/kmtrace.README.Debian	Wed Dec 10 07:08:45 2003
+++ kdesdk/debian/kmtrace.README.Debian	Thu Jan  1 01:00:00 1970
@@ -1,7 +0,0 @@
-KMtrace for Debian
-------------------
-
-The utility /usr/bin/match that is shipped with KMtrace has been renamed
-to /usr/bin/kmmatch in Debian to avoid conflicts with unrelated packages.
-
- -- Ben Burton <bab@debian.org>, Wed, 26 Feb 2003 18:52:09 +1100
diff -urN -x CVS kdesdk.orig/debian/kmtrace.install kdesdk/debian/kmtrace.install
--- kdesdk.orig/debian/kmtrace.install	Wed Feb 26 06:11:58 2003
+++ kdesdk/debian/kmtrace.install	Fri Oct 15 08:46:17 2004
@@ -3,5 +3,5 @@
 usr/bin/kmmatch
 usr/bin/kmtrace
 usr/include/kde/ktrace.h
-usr/lib/libktrace*
+usr/lib/kmtrace
 usr/share/apps/kmtrace
diff -urN -x CVS kdesdk.orig/debian/kmtrace.override kdesdk/debian/kmtrace.override
--- kdesdk.orig/debian/kmtrace.override	Mon May 13 13:33:33 2002
+++ kdesdk/debian/kmtrace.override	Thu Jan  1 01:00:00 1970
@@ -1,2 +0,0 @@
-# There is no -dev package.
-kmtrace: non-dev-pkg-with-shlib-symlink usr/lib/libktrace.so.1.0.0 usr/lib/libktrace.so
diff -urN -x CVS kdesdk.orig/debian/kmtrace.shlibs kdesdk/debian/kmtrace.shlibs
--- kdesdk.orig/debian/kmtrace.shlibs	Mon Sep 23 06:50:40 2002
+++ kdesdk/debian/kmtrace.shlibs	Thu Jan  1 01:00:00 1970
@@ -1 +0,0 @@
-libktrace 1 kmtrace (>= 3.0.8-0)
diff -urN -x CVS kdesdk.orig/debian/kompare.1 kdesdk/debian/kompare.1
--- kdesdk.orig/debian/kompare.1	Sun Feb  1 08:06:59 2004
+++ kdesdk/debian/kompare.1	Sat Oct 16 02:05:48 2004
@@ -2,7 +2,7 @@
 .\" First parameter, NAME, should be all caps
 .\" Second parameter, SECTION, should be 1-8, maybe w/ subsection
 .\" other parameters are allowed: see man(7), man(1)
-.TH KOMPARE 1 "January 31, 2004"
+.TH KOMPARE 1 "October 15, 2004"
 .\" Please adjust this date whenever revising the manpage.
 .\"
 .\" Some roff macros, for reference:
@@ -72,8 +72,8 @@
 The original documentation from the program author
 is available in \fI/usr/share/doc/kompare/\fP.
 .SH AUTHOR
-Kompare was written by John Firebaugh <jfirebaugh@kde.org> and
-Otto Bruggeman <otto.bruggeman@home.nl>.
+Kompare was written by John Firebaugh <jfirebaugh@kde.org>,
+Otto Bruggeman <otto.bruggeman@home.nl> and others.
 .br
 This manual page was prepared by Ben Burton <bab@debian.org>
 for the Debian GNU/Linux system (but may be used by others).
diff -urN -x CVS kdesdk.orig/debian/kompare.install kdesdk/debian/kompare.install
--- kdesdk.orig/debian/kompare.install	Mon Aug  2 00:38:28 2004
+++ kdesdk/debian/kompare.install	Fri Oct 15 08:46:17 2004
@@ -1,5 +1,6 @@
 usr/bin/kompare
-usr/lib/libkompareinterface.so.*
+usr/lib/libkompareinterface.so.0
+usr/lib/libkompareinterface.so.0.0.0
 usr/lib/kde3/libdialogpages.la
 usr/lib/kde3/libdialogpages.so*
 usr/lib/kde3/libkompare*
diff -urN -x CVS kdesdk.orig/debian/libcvsservice0.shlibs kdesdk/debian/libcvsservice0.shlibs
--- kdesdk.orig/debian/libcvsservice0.shlibs	Sat Jan 31 11:41:56 2004
+++ kdesdk/debian/libcvsservice0.shlibs	Fri Oct 15 08:46:17 2004
@@ -1 +1 @@
-libcvsservice 0 libcvsservice0 (>= 4:3.1.95-0)
+libcvsservice 0 libcvsservice0 (>= 4:3.3.0-0)
diff -urN -x CVS kdesdk.orig/debian/memprof2calltree.1 kdesdk/debian/memprof2calltree.1
--- kdesdk.orig/debian/memprof2calltree.1	Thu Jan  1 01:00:00 1970
+++ kdesdk/debian/memprof2calltree.1	Sat Oct 16 03:01:38 2004
@@ -0,0 +1,36 @@
+.\"                                      Hey, EMACS: -*- nroff -*-
+.\" First parameter, NAME, should be all caps
+.\" Second parameter, SECTION, should be 1-8, maybe w/ subsection
+.\" other parameters are allowed: see man(7), man(1)
+.TH MEMPROF2CALLTREE 1 "October 15, 2004"
+.\" Please adjust this date whenever revising the manpage.
+.\"
+.\" Some roff macros, for reference:
+.\" .nh        disable hyphenation
+.\" .hy        enable hyphenation
+.\" .ad l      left justify
+.\" .ad b      justify to both left and right margins
+.\" .nf        disable filling
+.\" .fi        enable filling
+.\" .br        insert line break
+.\" .sp <n>    insert n+1 empty lines
+.\" for manpage-specific macros, see man(7)
+.SH NAME
+memprof2calltree \- convert memprof profiling data to KCachegrind calltree format
+.SH SYNOPSIS
+.B memprof2calltree
+.SH DESCRIPTION
+\fBmemprof2calltree\fP reads the memory profiles of memprof and
+converts them into calltree format for use by KCachegrind.
+.PP
+The memprof data must be supplied on standard input, and the
+corresponding calltree data will be written to standard output.
+.PP
+This utility is part of the KDE Software Development Kit.
+.SH SEE ALSO
+.BR kcachegrind (1).
+.SH AUTHOR
+This converter was written by Josef Weidendorfer <Josef.Weidendorfer@gmx.de>.
+.br
+This manual page was prepared by Ben Burton <bab@debian.org>
+for the Debian GNU/Linux system (but may be used by others).
diff -urN -x CVS kdesdk.orig/debian/op2calltree.1 kdesdk/debian/op2calltree.1
--- kdesdk.orig/debian/op2calltree.1	Thu Jan  1 01:00:00 1970
+++ kdesdk/debian/op2calltree.1	Sat Oct 16 03:01:38 2004
@@ -0,0 +1,43 @@
+.\"                                      Hey, EMACS: -*- nroff -*-
+.\" First parameter, NAME, should be all caps
+.\" Second parameter, SECTION, should be 1-8, maybe w/ subsection
+.\" other parameters are allowed: see man(7), man(1)
+.TH OP2CALLTREE 1 "October 15, 2004"
+.\" Please adjust this date whenever revising the manpage.
+.\"
+.\" Some roff macros, for reference:
+.\" .nh        disable hyphenation
+.\" .hy        enable hyphenation
+.\" .ad l      left justify
+.\" .ad b      justify to both left and right margins
+.\" .nf        disable filling
+.\" .fi        enable filling
+.\" .br        insert line break
+.\" .sp <n>    insert n+1 empty lines
+.\" for manpage-specific macros, see man(7)
+.SH NAME
+op2calltree \- convert OProfile profiling data to KCachegrind calltree format
+.SH SYNOPSIS
+.B op2calltree
+.SH DESCRIPTION
+\fBop2calltree\fP reads the profiling data output by OProfile v0.8 and
+converts it into calltree format for use by KCachegrind.
+.PP
+The OProfile data must be supplied on standard input, and should be
+obtained by running:
+.PP
+.RS
+opreport \-gdf
+.RE
+.PP
+This converter will then write a separate calltree data file for each
+application, to a file of the form \fIoprof.out.<app>\fP.
+.PP
+This utility is part of the KDE Software Development Kit.
+.SH SEE ALSO
+.BR kcachegrind (1).
+.SH AUTHOR
+This converter was written by Josef Weidendorfer <Josef.Weidendorfer@gmx.de>.
+.br
+This manual page was prepared by Ben Burton <bab@debian.org>
+for the Debian GNU/Linux system (but may be used by others).
diff -urN -x CVS kdesdk.orig/debian/package_crystalsvg.1 kdesdk/debian/package_crystalsvg.1
--- kdesdk.orig/debian/package_crystalsvg.1	Thu Jan  1 01:00:00 1970
+++ kdesdk/debian/package_crystalsvg.1	Fri Oct 15 16:35:50 2004
@@ -0,0 +1,41 @@
+.\"                                      Hey, EMACS: -*- nroff -*-
+.\" First parameter, NAME, should be all caps
+.\" Second parameter, SECTION, should be 1-8, maybe w/ subsection
+.\" other parameters are allowed: see man(7), man(1)
+.TH PACKAGE_CRYSTALSVG 1 "October 15, 2004"
+.\" Please adjust this date whenever revising the manpage.
+.\"
+.\" Some roff macros, for reference:
+.\" .nh        disable hyphenation
+.\" .hy        enable hyphenation
+.\" .ad l      left justify
+.\" .ad b      justify to both left and right margins
+.\" .nf        disable filling
+.\" .fi        enable filling
+.\" .br        insert line break
+.\" .sp <n>    insert n+1 empty lines
+.\" for manpage-specific macros, see man(7)
+.SH NAME
+package_crystalsvg \- bundle up Crystal SVG sources from a KDE checkout
+.SH SYNOPSIS
+.B package_crystalsvg
+.SH DESCRIPTION
+\fBpackage_crystalsvg\fP assumes that the current directory contains one
+or more checked-out KDE modules.  It searches these modules for Crystal
+SVG vector sources (.svgz or .svg.gz files)
+and packages them up into a tarball.  The tarball will also
+contain \fIFILES\fP (describing the contents of the tarball and which
+KDE modules/paths they were copied from) as well as some other bits and
+pieces.
+.PP
+Note that you must at least have kdelibs checked out for this utility to
+run.
+.PP
+The final tarball will be created in the current working directory.
+.PP
+This utility is part of the KDE Software Development Kit.
+.SH AUTHOR
+package_crystalsvg was written by Frans Englich <frans.englich@telia.com>.
+.br
+This manual page was prepared by Ben Burton <bab@debian.org>
+for the Debian GNU/Linux system (but may be used by others).
diff -urN -x CVS kdesdk.orig/debian/patches/env.diff kdesdk/debian/patches/env.diff
--- kdesdk.orig/debian/patches/env.diff	Thu Jan  1 01:00:00 1970
+++ kdesdk/debian/patches/env.diff	Fri Oct 15 08:46:18 2004
@@ -0,0 +1,32 @@
+--- kdesdk/scripts/kdelnk2desktop.py	1999/07/07 02:19:18	1.1
++++ kdesdk/scripts/kdelnk2desktop.py	2002/11/14 03:35:17
+@@ -1,4 +1,4 @@
+-#! /usr/bin/env python
++#!/usr/bin/python
+ 
+ import os, sys, string
+ 
+--- kdesdk/scripts/zonetab2pot.py	2002/09/19 08:59:32	1.1
++++ kdesdk/scripts/zonetab2pot.py	2002/11/14 03:35:17
+@@ -1,4 +1,4 @@
+-#!/usr/bin/env python
++#!/usr/bin/python
+ '''This script reads timezone list as its first argument
+ or from /usr/share/zoneinfo/zone.tab, and converts it
+ to a PO file template.
+--- kdesdk/kcachegrind/converters/hotshot2calltree	28 Apr 2004 19:35:54 -0000	1.1
++++ kdesdk/kcachegrind/converters/hotshot2calltree	15 Oct 2004 05:50:05 -0000
+@@ -1,4 +1,4 @@
+-#!/usr/bin/env python
++#!/usr/bin/python
+ # _*_ coding: latin1 _*_
+ 
+ #
+--- kdesdk/kcachegrind/converters/pprof2calltree	28 Apr 2004 19:35:54 -0000	1.1
++++ kdesdk/kcachegrind/converters/pprof2calltree	15 Oct 2004 05:50:05 -0000
+@@ -1,4 +1,4 @@
+-#!/usr/bin/env php
++#!/usr/bin/php
+ # Redistribution and use in source and binary forms, with or without
+ # modification, are permitted provided that the following conditions are met:
+ #
diff -urN -x CVS kdesdk.orig/debian/patches/kde-buildrc.diff kdesdk/debian/patches/kde-buildrc.diff
--- kdesdk.orig/debian/patches/kde-buildrc.diff	Thu Jan  1 01:00:00 1970
+++ kdesdk/debian/patches/kde-buildrc.diff	Fri Oct 15 08:57:45 2004
@@ -0,0 +1,12 @@
+--- kdesdk/scripts/kde-build	17 Mar 2004 01:54:37 -0000	1.107
++++ kdesdk/scripts/kde-build	15 Oct 2004 06:56:51 -0000
+@@ -869,6 +869,9 @@ The following error codes are returned b
+ 
+         cd ~/scripts/; vi ./kde-buildrc; ./kde-build
+ 
++On Debian systems, an example F<kde-buildrc> can be found in
++the directory /usr/share/doc/kdesdk-scripts/.
++
+ =head1 BUGS
+ 
+ Lots, mostly that the script wasn't written with portability in mind
diff -urN -x CVS kdesdk.orig/debian/patches/kmtrace.diff kdesdk/debian/patches/kmtrace.diff
--- kdesdk.orig/debian/patches/kmtrace.diff	Thu Jan  1 01:00:00 1970
+++ kdesdk/debian/patches/kmtrace.diff	Sat Oct 16 01:06:14 2004
@@ -0,0 +1,51 @@
+--- kdesdk/kmtrace/Makefile.am	28 Apr 2003 06:52:31 -0000	1.18
++++ kdesdk/kmtrace/Makefile.am	15 Oct 2004 03:50:07 -0000
+@@ -33,13 +33,15 @@ kmmatch_LDFLAGS = $(all_libraries)
+ 
+ bin_SCRIPTS = kminspector
+ 
+-lib_LIBRARIES = libktrace_s.a
++kmtracelibdir = $(libdir)/kmtrace
++
++kmtracelib_LIBRARIES = libktrace_s.a
+ libktrace_s_a_SOURCES = ktrace_s.c
+ 
+ ktrace_s.c: ktrace.c
+ 	$(LN_S) $< $@
+ 
+-lib_LTLIBRARIES = libktrace.la
++kmtracelib_LTLIBRARIES = libktrace.la
+ libktrace_la_LDFLAGS = -avoid-version
+ libktrace_la_SOURCES = ksotrace.cpp ktrace.c
+ libktrace_la_LIBADD = $(LIBDL)
+--- kdesdk/kmtrace/README	24 May 2002 12:45:53 -0000	1.13
++++ kdesdk/kmtrace/README	15 Oct 2004 03:50:07 -0000
+@@ -23,7 +23,7 @@ friends.
+ 
+ You can activate malloc logging by starting yourApplication as:
+ 
+-	MALLOC_TRACE=/tmp/ktrace.out LD_PRELOAD=$KDEDIR/lib/libktrace.so yourApplication
++	MALLOC_TRACE=/tmp/ktrace.out LD_PRELOAD=/usr/lib/kmtrace/libktrace.so yourApplication
+ 
+ 2) The manual way
+ 
+@@ -37,7 +37,7 @@ Add as first statement in main():
+ 
+ Add ktrace_s.a to the LDADD line in your Makefile.am like:
+ 
+-	kicker_LDADD = kicker.la /opt/kde/lib/libktrace_s.a
++	kicker_LDADD = kicker.la /usr/lib/kmtrace/libktrace_s.a
+ 
+ Note that the static library is used.
+ You can now activate malloc logging by starting yourApplication as:
+--- kdesdk/kmtrace/kminspector.in	27 Nov 2001 01:44:01 -0000	1.2
++++ kdesdk/kmtrace/kminspector.in	15 Oct 2004 23:05:03 -0000
+@@ -2,7 +2,7 @@
+ 
+ export MALLOC_TREE=kminspector.tree
+ export MALLOC_THRESHOLD=2000
+-export LD_PRELOAD=@kde_libraries@/libktrace.so
++export LD_PRELOAD=@kde_libraries@/kmtrace/libktrace.so
+ 
+ $*
+ 
diff -urN -x CVS kdesdk.orig/debian/patches/pod2man.diff kdesdk/debian/patches/pod2man.diff
--- kdesdk.orig/debian/patches/pod2man.diff	Sun Aug  1 16:38:01 2004
+++ kdesdk/debian/patches/pod2man.diff	Fri Oct 15 08:46:18 2004
@@ -4,7 +4,7 @@
  #%.1: %
  #	pod2man $< $@
  
-+pod2man_options = "--center=K Desktop Environment (SDK)" "--release=KDE 3.2"
++pod2man_options = "--center=K Desktop Environment (SDK)" "--release=KDE 3.3"
 +
  kde-build.1: kde-build
 -	pod2man $< > $@
diff -urN -x CVS kdesdk.orig/debian/patches/python.diff kdesdk/debian/patches/python.diff
--- kdesdk.orig/debian/patches/python.diff	Thu Nov 14 04:38:48 2002
+++ kdesdk/debian/patches/python.diff	Thu Jan  1 01:00:00 1970
@@ -1,16 +0,0 @@
---- kdesdk/scripts/kdelnk2desktop.py	1999/07/07 02:19:18	1.1
-+++ kdesdk/scripts/kdelnk2desktop.py	2002/11/14 03:35:17
-@@ -1,4 +1,4 @@
--#! /usr/bin/env python
-+#!/usr/bin/python
- 
- import os, sys, string
- 
---- kdesdk/scripts/zonetab2pot.py	2002/09/19 08:59:32	1.1
-+++ kdesdk/scripts/zonetab2pot.py	2002/11/14 03:35:17
-@@ -1,4 +1,4 @@
--#!/usr/bin/env python
-+#!/usr/bin/python
- '''This script reads timezone list as its first argument
- or from /usr/share/zoneinfo/zone.tab, and converts it
- to a PO file template.
diff -urN -x CVS kdesdk.orig/debian/png2mng.pl.1 kdesdk/debian/png2mng.pl.1
--- kdesdk.orig/debian/png2mng.pl.1	Thu Jan  1 01:00:00 1970
+++ kdesdk/debian/png2mng.pl.1	Fri Oct 15 16:35:50 2004
@@ -0,0 +1,49 @@
+.\"                                      Hey, EMACS: -*- nroff -*-
+.\" First parameter, NAME, should be all caps
+.\" Second parameter, SECTION, should be 1-8, maybe w/ subsection
+.\" other parameters are allowed: see man(7), man(1)
+.TH PNG2MNG.PL 1 "October 15, 2004"
+.\" Please adjust this date whenever revising the manpage.
+.\"
+.\" Some roff macros, for reference:
+.\" .nh        disable hyphenation
+.\" .hy        enable hyphenation
+.\" .ad l      left justify
+.\" .ad b      justify to both left and right margins
+.\" .nf        disable filling
+.\" .fi        enable filling
+.\" .br        insert line break
+.\" .sp <n>    insert n+1 empty lines
+.\" for manpage-specific macros, see man(7)
+.SH NAME
+png2mng.pl \- combine a set of PNG images into an animated MNG image
+.SH SYNOPSIS
+.B png2mng.pl
+.I basename width height
+.SH DESCRIPTION
+\fBpng2mng.pl\fP reads a sequence of PNG images and combines them into
+a single animated MNG image.
+.PP
+The individual PNG images should be numbered consecutively from zero up,
+and should have names of the form
+.I basenameXXXX
+where
+.I XXXX
+is the four-digit sequence number (0000, 0001, 0002, etc).
+The width and height of the PNG images should
+also be included on the command line.
+.PP
+The final animated MNG image will be written to standard output.
+.PP
+This utility is part of the KDE Software Development Kit.
+.SH EXAMPLE
+The following example will combine the files hi32-action-kde-0000.png,
+hi32-action-kde-0001.png, hi32-action-kde-0002.png, etc., and write
+the resulting animated image to hi32-action-kde.mng:
+.PP
+.RS
+png2mng.pl hi32-action-kde- 32 32 > hi32-action-kde.mng
+.RE
+.SH AUTHOR
+This manual page was prepared by Ben Burton <bab@debian.org>
+for the Debian GNU/Linux system (but may be used by others).
diff -urN -x CVS kdesdk.orig/debian/pprof2calltree.1 kdesdk/debian/pprof2calltree.1
--- kdesdk.orig/debian/pprof2calltree.1	Thu Jan  1 01:00:00 1970
+++ kdesdk/debian/pprof2calltree.1	Sat Oct 16 03:01:38 2004
@@ -0,0 +1,52 @@
+.\"                                      Hey, EMACS: -*- nroff -*-
+.\" First parameter, NAME, should be all caps
+.\" Second parameter, SECTION, should be 1-8, maybe w/ subsection
+.\" other parameters are allowed: see man(7), man(1)
+.TH PPROF2CALLTREE 1 "October 15, 2004"
+.\" Please adjust this date whenever revising the manpage.
+.\"
+.\" Some roff macros, for reference:
+.\" .nh        disable hyphenation
+.\" .hy        enable hyphenation
+.\" .ad l      left justify
+.\" .ad b      justify to both left and right margins
+.\" .nf        disable filling
+.\" .fi        enable filling
+.\" .br        insert line break
+.\" .sp <n>    insert n+1 empty lines
+.\" for manpage-specific macros, see man(7)
+.SH NAME
+pprof2calltree \- convert APD profiling data to KCachegrind calltree format
+.SH SYNOPSIS
+.B pprof2calltree
+\fB\-f\fP \fIinput\fP
+.SH DESCRIPTION
+\fBpprof2calltree\fP reads the pprof profiling data output by
+APD (http://pecl.php.net/apd/), and converts it into calltree format
+for use by KCachegrind.
+.PP
+This utility is part of the KDE Software Development Kit.
+.SH OPTIONS
+.TP
+\fB\-f\fP \fIinput\fP
+Specifies the APD pprof trace file to convert.  This generally has a
+name of the form \fIpprof.XXXXX.Y\fP.  The calltree output for
+KCachegrind will then be written to the file \fIcachegrind.out.XXXXX.Y\fP.
+.SH EXAMPLE
+Suppose you profile a script with APD and obtain the pprof trace file
+\fIpprof.12345.1\fP.  You can then work with this data in KCachegrind as
+follows.
+.PP
+.RS
+pprof2calltree -f pprof.12345.1
+.br
+kcachegrind cachegrind.out.12345.1
+.RE
+.PP
+.SH SEE ALSO
+.BR kcachegrind (1).
+.SH AUTHOR
+This converter was written by George Schlossnagle <george@omniti.com>.
+.br
+This manual page was prepared by Ben Burton <bab@debian.org>
+for the Debian GNU/Linux system (but may be used by others).
diff -urN -x CVS kdesdk.orig/debian/rules kdesdk/debian/rules
--- kdesdk.orig/debian/rules	Sun Aug  1 10:29:31 2004
+++ kdesdk/debian/rules	Sat Oct 16 03:46:14 2004
@@ -45,6 +45,10 @@
 configure-stamp:
 	dh_testdir
 
+	# Make absolutely sure these aren't left around from the .orig.tar.gz.
+	-rm -f kmtrace.shlibs kmtrace.overrides kmtrace.README.Debian
+	-rm -f patches/python.diff
+
 	# Apply Debian specific patches
 	if test ! -f patch-stamp; then \
 		for patch in $(CURDIR)/debian/patches/*.diff ;\
@@ -140,11 +144,10 @@
 	$(INSTALL_FILE) scheck/README \
 	  debian/tmp/usr/share/doc/kdesdk-misc/README.scheck
 
-	# Rename /usr/bin/match to /usr/bin/kmmatch.
-	if test -e debian/tmp/usr/bin/match; then \
-		mv debian/tmp/usr/bin/match debian/tmp/usr/bin/kmmatch; \
-	fi
-	
+	$(INSTALL_DIR) debian/tmp/usr/share/doc/kdesdk-scripts
+	$(INSTALL_FILE) scripts/kde-buildrc \
+	  debian/tmp/usr/share/doc/kdesdk-scripts/kde-buildrc.example
+
 	# Split into separate binary packages.
 	dh_install --sourcedir=debian/tmp --list-missing
 
@@ -203,12 +206,17 @@
 	for pkg in `dh_listpackages`; do \
 	  case "$$pkg" in \
 	    kdesdk-scripts ) \
+	      mv debian/kdesdk-scripts/usr/share/man/pl/man1/kdemangen.1 \
+	        debian/kdesdk-scripts/usr/share/man/man1/kdemangen.pl.1; \
+	      mv debian/kdesdk-scripts/usr/share/man/pl/man1/png2mng.1 \
+	        debian/kdesdk-scripts/usr/share/man/man1/png2mng.pl.1; \
 	      mv debian/kdesdk-scripts/usr/share/man/py/man1/kdelnk2desktop.1 \
 	        debian/kdesdk-scripts/usr/share/man/man1/kdelnk2desktop.py.1; \
 	      mv debian/kdesdk-scripts/usr/share/man/py/man1/zonetab2pot.1 \
 	        debian/kdesdk-scripts/usr/share/man/man1/zonetab2pot.py.1; \
 	      mv debian/kdesdk-scripts/usr/share/man/sh/man1/build-progress.1 \
 	        debian/kdesdk-scripts/usr/share/man/man1/build-progress.sh.1; \
+	      rm -rf debian/kdesdk-scripts/usr/share/man/pl; \
 	      rm -rf debian/kdesdk-scripts/usr/share/man/py; \
 	      rm -rf debian/kdesdk-scripts/usr/share/man/sh; \
 	    ;; \
diff -urN -x CVS kdesdk.orig/debian/source.lintian-overrides kdesdk/debian/source.lintian-overrides
--- kdesdk.orig/debian/source.lintian-overrides	Sun Aug  1 10:29:31 2004
+++ kdesdk/debian/source.lintian-overrides	Wed Oct 13 03:35:15 2004
@@ -1,2 +1,2 @@
-kdesdk source: cvsignore-file-in-source
-kdesdk source: source-contains-CVS-dir
+cvsignore-file-in-source
+source-contains-CVS-dir
diff -urN -x CVS kdesdk.orig/debian/umbrello.xpm kdesdk/debian/umbrello.xpm
--- kdesdk.orig/debian/umbrello.xpm	Sat Jan 31 11:41:56 2004
+++ kdesdk/debian/umbrello.xpm	Fri Oct 15 16:35:50 2004
@@ -1,45 +1,103 @@
 /* XPM */
-static char *umbrello[] = {
-/* columns rows colors chars-per-pixel */
-"32 32 7 1",
-"  c black",
-". c red",
-"X c #FF4030",
-"o c #FF8060",
-"O c #FFBF90",
-"+ c #FFFFC0",
-"@ c None",
-/* pixels */
-"@@.......@@@@@@@@@@@@@@@@@@@@@@@",
-"@@.......@@@@@@@@@@@@@@@@@@@@@@@",
-"....XoX....@@@@@@@@@@@@@@@@@@@@@",
-"....o+o....@@@@@@@@@@@@@@@@@@@@@",
-"..XoO+OoX..@@@@@@@@@@@@@@@@@@@@@",
-"..o+++++o..@@@@@@@@@@@@@@@@@@@@@",
-"..XoO+OoX..@@@@@@@@@@@@@@@@@@@@@",
-"....o+o....@@@@@@@@@@@@@@@@@@@@@",
-"....XoX......@@@@@@@@@@@@@@@@@@@",
-"@@.......@...@@@@@@@@@@@@@@@@@@@",
-"@@.......@.....@@@@@@@@@@@@@@@@@",
-"@@@@...@@@@@...@@@@@@@@@@@@@@@@@",
-"...........@.....@@@@@@@@@@@@@@@",
-"...........@@@...@@@@@@@@@@@@@@@",
-"...........@@@.....@@@@@@@@@@@@@",
-"@@@@...@@@@@@@@@...@@@@@@@@@@@@@",
-"@@@@...@@@@@@@@@.....@@@@@@@@@@@",
-"@@@@...@@@@@@@@@@@...@@@@@@@@@@@",
-"@@.......@@@@@@@.............@@@",
-"@@.......@@@@@@@.............@@@",
-"...........@@@....XoooooooX....@",
-"...@@@@@...@@@....o+++++++o....@",
-"...@@@@@...@@@..XoO+++++++OoX..@",
-"...@@@@@...@@@..o+++++++++++o..@",
-"...@@@@@...@@@..o+++++++++++o..@",
-"@@@@@@@@@@@@@@..o+++++++++++o..@",
-"@@@@@@@@@@@@@@..XoO+++++++OoX..@",
-"@@@@@@@@@@@@@@....o+++++++o....@",
-"@@@@@@@@@@@@@@....XoooooooX....@",
-"@@@@@@@@@@@@@@@@.............@@@",
-"@@@@@@@@@@@@@@@@.............@@@",
-"@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@"
-};
+static char * umbrello_xpm[] = {
+"32 32 68 1",
+" 	c None",
+".	c #FF0000",
+"+	c #FF0100",
+"@	c #FF0101",
+"#	c #FF0503",
+"$	c #FF0605",
+"%	c #FF0E0A",
+"&	c #FF100C",
+"*	c #FF140F",
+"=	c #FF1712",
+"-	c #FF1813",
+";	c #FF1913",
+">	c #FF1F17",
+",	c #FF1F18",
+"'	c #FF2019",
+")	c #FF221A",
+"!	c #FF241C",
+"~	c #FF261D",
+"{	c #FF291F",
+"]	c #FF2A20",
+"^	c #FF2C22",
+"/	c #FF3025",
+"(	c #FF3629",
+"_	c #FF372B",
+":	c #FF4233",
+"<	c #FF4837",
+"[	c #FF4938",
+"}	c #FF4939",
+"|	c #FF4C3B",
+"1	c #FF4E3C",
+"2	c #FF5945",
+"3	c #FF5E48",
+"4	c #FF604A",
+"5	c #FF624C",
+"6	c #FF6850",
+"7	c #FF745A",
+"8	c #FF775C",
+"9	c #FF7B60",
+"0	c #FF7F63",
+"a	c #FF8C6D",
+"b	c #FF8F6F",
+"c	c #FF9775",
+"d	c #FF9976",
+"e	c #FFA07C",
+"f	c #FFAD86",
+"g	c #FFAF87",
+"h	c #FFB38B",
+"i	c #FFB98F",
+"j	c #FFBC92",
+"k	c #FFC498",
+"l	c #FFC89B",
+"m	c #FFCFA1",
+"n	c #FFD3A3",
+"o	c #FFD9A8",
+"p	c #FFDBAA",
+"q	c #FFDFAD",
+"r	c #FFE2AF",
+"s	c #FFE8B4",
+"t	c #FFEBB6",
+"u	c #FFEDB8",
+"v	c #FFEFB9",
+"w	c #FFEFBA",
+"x	c #FFF4BD",
+"y	c #FFF9C1",
+"z	c #FFFBC2",
+"A	c #FFFDC4",
+"B	c #FFFEC5",
+"C	c #FFFFC6",
+"    ...                         ",
+"  .......                       ",
+" +.}hme!..                      ",
+"..2CCCCw'.                      ",
+"..lCCCCC0.                      ",
+"..kCCCCC9.                      ",
+" .|BCCCs;.                      ",
+" @._eja-...                     ",
+"  ..........                    ",
+"    ...   ...                   ",
+"    ..     ...                  ",
+"    ..      ...                 ",
+"...........  ...                ",
+"...........   ...               ",
+"..........      ...             ",
+"    ..           ...            ",
+"    ..            ...           ",
+"    ...            ...          ",
+"   .....            ...         ",
+"  ... ..         ..........     ",
+"  ..  ...      ......=>#.....   ",
+" ..    ...    .../dpyCCArg<...  ",
+"...     ..   ..&iCCCCCCCCCCo].. ",
+"             .%nCCCCCCCCCCCCu^..",
+"            ..3CCCCCCCCCCCCCCc..",
+"            ..7CCCCCCCCCCCCCCf..",
+"             .,xCCCCCCCCCCCCC1..",
+"             ..(tCCCCCCCCCCz4.. ",
+"              ..*8qCCCCCCvb~..  ",
+"               ...@):56[{$...   ",
+"                 ..........     ",
+"                     ..         "};
diff -urN -x CVS kdesdk.orig/kbabel/catalogmanager/catalogmanager.cpp kdesdk/kbabel/catalogmanager/catalogmanager.cpp
--- kdesdk.orig/kbabel/catalogmanager/catalogmanager.cpp	Wed Aug  4 18:13:57 2004
+++ kdesdk/kbabel/catalogmanager/catalogmanager.cpp	Thu Oct  7 18:41:02 2004
@@ -456,7 +456,11 @@
     action = new KAction(i18n("&Delete"),Key_Delete,_catalogManager,SLOT(slotDeleteFile()),actionCollection(), "delete");
     action->setEnabled(false);
 
+#if KDE_IS_VERSION( 3, 2, 90 )
     setupGUI();
+#else
+    createGUI();
+#endif
 }
 
 void CatalogManager::setupStatusBar()
diff -urN -x CVS kdesdk.orig/kbabel/common/kbabelfilter.desktop kdesdk/kbabel/common/kbabelfilter.desktop
--- kdesdk.orig/kbabel/common/kbabelfilter.desktop	Fri Sep 17 15:49:17 2004
+++ kdesdk/kbabel/common/kbabelfilter.desktop	Tue Oct  5 15:20:58 2004
@@ -3,6 +3,7 @@
 Type=ServiceType
 X-KDE-ServiceType=KBabelFilter
 Comment=KBabel filter
+Comment[br]=Sil KBabel
 Comment[ca]=Filtre de KBabel
 Comment[cs]=KBabel filtr
 Comment[cy]=Hidlydd KBabel 
diff -urN -x CVS kdesdk.orig/kbabel/commonui/kactionselector.cpp kdesdk/kbabel/commonui/kactionselector.cpp
--- kdesdk.orig/kbabel/commonui/kactionselector.cpp	Tue Nov 25 23:32:22 2003
+++ kdesdk/kbabel/commonui/kactionselector.cpp	Mon Oct 18 13:28:49 2004
@@ -117,6 +117,7 @@
 
 KActionSelector::~KActionSelector()
 {
+		delete d;
 }
 
 //END Constructor/destroctor
diff -urN -x CVS kdesdk.orig/kbabel/commonui/projectwizardwidget.ui kdesdk/kbabel/commonui/projectwizardwidget.ui
--- kdesdk.orig/kbabel/commonui/projectwizardwidget.ui	Fri Oct  1 16:44:48 2004
+++ kdesdk/kbabel/commonui/projectwizardwidget.ui	Thu Oct  7 18:41:02 2004
@@ -1,4 +1,4 @@
-<!DOCTYPE UI><UI version="3.3" stdsetdef="1">
+<!DOCTYPE UI><UI version="3.2" stdsetdef="1">
 <class>ProjectStep1</class>
 <author>Stanislav Visnovsky &lt;visnovsky@kde.org&gt;</author>
 <widget class="QWidget">
diff -urN -x CVS kdesdk.orig/kbabel/commonui/projectwizardwidget2.ui kdesdk/kbabel/commonui/projectwizardwidget2.ui
--- kdesdk.orig/kbabel/commonui/projectwizardwidget2.ui	Tue Jun 22 23:16:48 2004
+++ kdesdk/kbabel/commonui/projectwizardwidget2.ui	Thu Oct  7 18:41:02 2004
@@ -1,4 +1,4 @@
-<!DOCTYPE UI><UI version="3.3" stdsetdef="1">
+<!DOCTYPE UI><UI version="3.2" stdsetdef="1">
 <class>ProjectStep2</class>
 <widget class="QWidget">
     <property name="name">
diff -urN -x CVS kdesdk.orig/kbabel/datatools/length/kbabel_lengthtool.desktop kdesdk/kbabel/datatools/length/kbabel_lengthtool.desktop
--- kdesdk.orig/kbabel/datatools/length/kbabel_lengthtool.desktop	Fri Sep 17 15:49:19 2004
+++ kdesdk/kbabel/datatools/length/kbabel_lengthtool.desktop	Tue Oct  5 15:20:58 2004
@@ -18,9 +18,10 @@
 Name[pt]=Validador do Tamanho da Mensagem Traduzida para o KBabel
 Name[pt_BR]=Um Validador de Tamanho de Mensagem Traduzida para o KBabel
 Name[ru]=Утилита проверки длины переведённых сообщений для KBabel
+Name[sk]=Kontrola dĺžky preložených textov pre KBabel
 Name[sl]=Potrjevalnik dolžin prevedenih sporočil za KBabel
 Name[sr]=Оверивач дужине преведених порука за KBabel
-Name[sr@Latn]=Оверивач дужине преведених порука за KBabel
+Name[sr@Latn]=Overivač dužine prevedenih poruka za KBabel
 Name[sv]=Validering av meddelandelängd för översättningar i Kbabel
 Name[ta]= Kபாபேலுக்கான மொழிபெயர்ப்பு செய்தி அளவு மதிப்பீட்டாளர்
 Name[tg]=Утилитаи тафтиши дарозии ахборҳои тарҷумакунанда барои KBabel
diff -urN -x CVS kdesdk.orig/kbabel/datatools/not-translated/kbabel_nottranslatedtool.desktop kdesdk/kbabel/datatools/not-translated/kbabel_nottranslatedtool.desktop
--- kdesdk.orig/kbabel/datatools/not-translated/kbabel_nottranslatedtool.desktop	Fri Sep 17 15:49:19 2004
+++ kdesdk/kbabel/datatools/not-translated/kbabel_nottranslatedtool.desktop	Tue Oct  5 15:20:59 2004
@@ -18,9 +18,10 @@
 Name[pt]=Verificação de Mensagens Traduzidas que Contenham Inglês para o KBabel
 Name[pt_BR]=Verifica por Strings Traduzidos Contendo Inglês para o KBabel
 Name[ru]=Проверка переведённых строк на равенство оригинальным для KBabel
+Name[sk]=Kontrola anglického textu v preklade pre KBabel
 Name[sl]=Preveri prevedene nize, ki vsebujejo angleščino; za KBabel
 Name[sr]=Провера за преведене низове знакова који садрже енглески за KBabel
-Name[sr@Latn]=Провера за преведене низове знакова који садрже енглески за KBabel
+Name[sr@Latn]=Provera za prevedene nizove znakova koji sadrže engleski za KBabel
 Name[sv]=Kontrollera översatta strängar som innehåller engelska för Kbabel
 Name[ta]=கேபாபேலுக்கான ஆங்கில மொழிபெயர்ப்பு சரங்களை சோதிக்கவும். 
 Name[tg]=Тафтиши сатрҳои тарҷумашуда ба баробарии аслӣ барои KBabel
diff -urN -x CVS kdesdk.orig/kbabel/datatools/whitespace/kbabel_whitespacetool.desktop kdesdk/kbabel/datatools/whitespace/kbabel_whitespacetool.desktop
--- kdesdk.orig/kbabel/datatools/whitespace/kbabel_whitespacetool.desktop	Fri Sep 17 15:49:20 2004
+++ kdesdk/kbabel/datatools/whitespace/kbabel_whitespacetool.desktop	Tue Oct  5 15:20:59 2004
@@ -18,9 +18,10 @@
 Name[pt]=Validação de Texto Traduzido como Espaço para o KBabel
 Name[pt_BR]=Validador de String Traduzidos com Espaços em Branco para o KBabel
 Name[ru]=Проверка строк, переведённых как пробел для KBabel
+Name[sk]=Kontrola textu preloženého ako prázdny pre KBabel
 Name[sl]=Potrjevalnik nizov, prevedenih kot presledki, za KBabel
 Name[sr]=Оверивач за низове знакова преведене као празан простор за KBabel
-Name[sr@Latn]=Оверивач за низове знакова преведене као празан простор за KBabel
+Name[sr@Latn]=Overivač za nizove znakova prevedene kao prazan prostor za KBabel
 Name[sv]=Validering av sträng översatt som blanktecken för Kbabel
 Name[ta]=Kபாபேலுக்கு சரம் மொழிபெயர்ப்பு இடைவேலை மதிப்பீட்டாளர்
 Name[tg]=Тафтиши сатр, ки мисли фосила барои KBabel, тарҷума шудааст
diff -urN -x CVS kdesdk.orig/kbabel/kbabel/main.cpp kdesdk/kbabel/kbabel/main.cpp
--- kdesdk.orig/kbabel/kbabel/main.cpp	Sun Jul 18 19:21:10 2004
+++ kdesdk/kbabel/kbabel/main.cpp	Tue Oct  5 07:44:18 2004
@@ -106,6 +106,7 @@
     if (!isRestored() || !first)
 #endif
     {
+	kdDebug () << "Suspending DCOP" << endl;
 	kapp->dcopClient()->suspend();
 
         KCmdLineArgs *args = KCmdLineArgs::parsedArgs();
@@ -138,6 +139,9 @@
         {
             kdDebug(KBABEL) << "gotomsgid" << endl;
             QString m = QString::fromLocal8Bit(msgid);
+
+	    kdDebug () << "Resuming DCOP" << endl;
+	    kapp->dcopClient()->resume();
             kbInterface->gotoFileEntry(args->url(0).url().utf8(),m.utf8());
         }
         else
@@ -164,6 +168,9 @@
             {
                widget->open( args->url(i) , QString::null, i != 0 );
             }
+
+	    kdDebug () << "Resuming DCOP" << endl;
+	    kapp->dcopClient()->resume();
         }
 
 
@@ -185,8 +192,6 @@
                     "Thanks.", "Warning");
             */
         }
-
-	kapp->dcopClient()->resume();
     }
 
     return 0;
@@ -209,6 +214,10 @@
     kdDebug() << "openURL " << url << endl;
 
     KURL u(QString::fromLocal8Bit(url));
+
+    kdDebug () << "Suspending DCOP" << endl;
+    kapp->dcopClient()->suspend();
+
     KBabelMW *kb = KBabelMW::winForURL(u,project);
     if(kb)
     {
@@ -227,6 +236,10 @@
         	kb = static_cast<KBabelMW*>(mw);
         	KWin::activateWindow(kb->topLevelWidget()->winId());
         	kb->open(u, QString::fromUtf8(package),newWindow);
+
+		kdDebug () << "Resuming DCOP" << endl;
+		kapp->dcopClient()->resume();
+
 		return ;
 	    }
 	    mw = KMainWindow::memberList->next();
@@ -241,6 +254,10 @@
     	    KWin::setActiveWindow(kb->topLevelWidget()->winId());
 	    kb->projectOpen(projectFile);
     	    kb->open(u,QString::fromUtf8(package),false);
+
+	    kdDebug () << "Resuming DCOP" << endl;
+	    kapp->dcopClient()->resume();
+
 	    return;
 	}
 
@@ -253,6 +270,10 @@
         	kb = static_cast<KBabelMW*>(mw);
         	KWin::activateWindow(kb->topLevelWidget()->winId());
         	kb->open(u, QString::fromUtf8(package),newWindow);
+
+		kdDebug () << "Resuming DCOP" << endl;
+		kapp->dcopClient()->resume();
+
 		return ;
 	    }
 	    mw = KMainWindow::memberList->next();
@@ -266,6 +287,9 @@
         KWin::activateWindow(kb->topLevelWidget()->winId());
         kb->open(u,QString::fromUtf8(package),newWindow);
     }
+    
+    kdDebug () << "Resuming DCOP" << endl;
+    kapp->dcopClient()->resume();
 }
 
 void KBabelInterface::openTemplate(QCString openFilename, QCString saveFilename, QCString package, int newWindow)
@@ -279,6 +303,10 @@
 
     KURL u(saveFilename.data());
     KURL t(openFilename.data());
+
+    kdDebug () << "Suspending DCOP" << endl;
+    kapp->dcopClient()->suspend();
+
     KBabelMW *kb = KBabelMW::winForURL(u, project);
     if(kb)
     {
@@ -305,6 +333,9 @@
             kb->openTemplate(t,u,QString::fromUtf8(package));
         }
     }
+
+    kdDebug () << "Resuming DCOP" << endl;
+    kapp->dcopClient()->resume();
 }
 
 void KBabelInterface::gotoFileEntry(QCString url, QCString m)
@@ -414,6 +445,9 @@
 
 KBabelMW* KBabelInterface::findInstance( QCString url, QCString project, QCString package)
 {
+    kdDebug () << "Suspending DCOP" << endl;
+    kapp->dcopClient()->suspend();
+
     KURL u ( url );
     KBabelMW *kb = 0;
     if( !url.isNull() )
@@ -441,6 +475,10 @@
         kb->show();
         if( !url.isEmpty() ) kb->open(u,QString::fromUtf8(package),false);
     }
+
+    kdDebug () << "Resuming DCOP" << endl;
+    kapp->dcopClient()->resume();
+
     return kb;
 }
 
diff -urN -x CVS kdesdk.orig/kbugbuster/gui/cwloadingwidget.cpp kdesdk/kbugbuster/gui/cwloadingwidget.cpp
--- kdesdk.orig/kbugbuster/gui/cwloadingwidget.cpp	Fri Dec 28 01:53:06 2001
+++ kdesdk/kbugbuster/gui/cwloadingwidget.cpp	Sun Oct  3 13:12:47 2004
@@ -58,9 +58,9 @@
     setFrameShape( StyledPanel );
     setFrameShadow( Sunken );
     setLineWidth( 2 );
-    
+
     setBackgroundMode( NoBackground ); // no flicker
-    
+
     // Load images and prepare pixmap effect
     if( m_mode == TopFrame )
     {
@@ -124,7 +124,7 @@
     {
         QFont bigFont = QFont( KGlobalSettings::generalFont().family(),
                                28, QFont::Bold, true );
-    
+
         // Draw logo
         p.drawPixmap( 0, 0, *m_logoPixmap );
 
@@ -144,12 +144,12 @@
         // Draw intro text
         QString desc = i18n( "Welcome to KBugBuster, a tool to manage the "
                              "KDE Bug Report System. With KBugBuster you can "
-                             "manage oustanding bug reports for KDE from a "
+                             "manage outstanding bug reports for KDE from a "
                              "convenient front end." );
         p.setPen( black );
         p.drawText( 28, 128, cr.width() - 28, 184 - 128,
                     AlignAuto | AlignVCenter | WordBreak, desc );
-                    
+
         // Draw the caption text
         QString caption = i18n( "KBugBuster" );
         p.setFont( bigFont );
@@ -184,11 +184,11 @@
         int boxW = cr.width() - 2 * boxX - 2 * 10;
         if( boxW > 500 )
             boxW = 500;
-        
+
         QRect br = fontMetrics().boundingRect( boxX, boxY,
                    boxW, cr.height() - boxY - 10 - 2 * fheight,
                    AlignAuto | AlignTop | WordBreak, m_text );
-        
+
         QRect box = br;
         box.setHeight( box.height() + 2 * fheight );
         box.setWidth( box.width() + 2 * 10 );
@@ -196,7 +196,7 @@
             box.setWidth( QMIN( cr.width() - 2 * boxX, 500 + 2 * 10 ) );
         if( box.height() < 100 )
             box.setHeight( QMIN( cr.height() - boxY - 2 * fheight - 10, 100 ) );
-        
+
         p.setClipRect( box );
         p.fillRect( box, QBrush( QColor( 204, 222, 234 ) ) );
         p.drawPixmap( toolsEffectX, toolsEffectY, *m_toolsPixmapEffect );
diff -urN -x CVS kdesdk.orig/kbugbuster/kresources/Makefile.am kdesdk/kbugbuster/kresources/Makefile.am
--- kdesdk.orig/kbugbuster/kresources/Makefile.am	Mon Feb  9 11:55:55 2004
+++ kdesdk/kbugbuster/kresources/Makefile.am	Sun Oct 10 16:35:53 2004
@@ -12,3 +12,5 @@
 
 METASOURCES = AUTO
 
+messages: rc.cpp
+	$(XGETTEXT) *.cpp -o $(podir)/kres_bugzilla.pot
diff -urN -x CVS kdesdk.orig/kbugbuster/kresources/bugzilla.desktop kdesdk/kbugbuster/kresources/bugzilla.desktop
--- kdesdk.orig/kbugbuster/kresources/bugzilla.desktop	Fri Sep 17 15:49:25 2004
+++ kdesdk/kbugbuster/kresources/bugzilla.desktop	Tue Oct  5 15:21:01 2004
@@ -18,9 +18,10 @@
 Name[pt]=Lista de Tarefas do Bugzilla
 Name[pt_BR]=Lista de Pendências do Bugzilla
 Name[ru]=Список Todo Bugzilla
+Name[sk]=Zoznam úloh v Bugzille
 Name[sl]=Seznam »za-narediti« v Bugzilli
 Name[sr]=Листа за-урадити Bugzilla-е
-Name[sr@Latn]=Листа за-урадити Bugzilla-е
+Name[sr@Latn]=Lista za-uraditi Bugzilla-e
 Name[sv]=Bugzilla att-göra lista
 Name[ta]=பக்சிலா டோடோஸ் பட்டியல்
 Name[tg]=Рӯйхати Todo Bugzilla
diff -urN -x CVS kdesdk.orig/kbugbuster/kresources/kcalresource_plugin.cpp kdesdk/kbugbuster/kresources/kcalresource_plugin.cpp
--- kdesdk.orig/kbugbuster/kresources/kcalresource_plugin.cpp	Fri Jan 23 16:36:56 2004
+++ kdesdk/kbugbuster/kresources/kcalresource_plugin.cpp	Sun Oct 10 16:35:53 2004
@@ -22,12 +22,16 @@
 #include "kcalresourceconfig.h"
 #include "kcalresource.h"
 
+#include <kglobal.h>
+#include <klocale.h>
+
 using namespace KCal;
 
 extern "C"
 {
   void *init_kcal_bugzilla()
   {
+    KGlobal::locale()->insertCatalogue( "kres_bugzilla" );
     return new KRES::PluginFactory<KCalResource,KCalResourceConfig>();
   }
 }
diff -urN -x CVS kdesdk.orig/kdeaccounts-plugin/kdeaccountsplugin.desktop kdesdk/kdeaccounts-plugin/kdeaccountsplugin.desktop
--- kdesdk.orig/kdeaccounts-plugin/kdeaccountsplugin.desktop	Fri Sep 17 15:49:29 2004
+++ kdesdk/kdeaccounts-plugin/kdeaccountsplugin.desktop	Sun Oct  3 15:38:31 2004
@@ -2,6 +2,7 @@
 Encoding=UTF-8
 Name=KDE CVS Accounts
 Name[af]=Kde Cvs Rekeninge
+Name[br]=Kontoù CVS KDE
 Name[bs]=KDE CVS računi
 Name[ca]=Comptes CVS de KDE
 Name[cs]=Účty z CVS KDE
diff -urN -x CVS kdesdk.orig/kuiviewer/kuiviewer.desktop kdesdk/kuiviewer/kuiviewer.desktop
--- kdesdk.orig/kuiviewer/kuiviewer.desktop	Fri Sep 17 15:49:35 2004
+++ kdesdk/kuiviewer/kuiviewer.desktop	Tue Oct  5 15:21:03 2004
@@ -28,9 +28,10 @@
 GenericName[pt]=Visualizador de Ficheiros UI do Qt Designer
 GenericName[pt_BR]=Visualizador de Arquivos e Interface do Qt Designer
 GenericName[ru]=Просмотрщик UI-файлов Qt Designer
+GenericName[sk]=Prehliadač súborov UI pre Qt Designer
 GenericName[sl]=Pregledovalnik datotek UI za Qt Designer
 GenericName[sr]=Приказивач UI фајлова Qt Designer-а
-GenericName[sr@Latn]=Приказивач UI фајлова Qt Designer-а
+GenericName[sr@Latn]=Prikazivač UI fajlova Qt Designer-a
 GenericName[sv]=Qt-designer UI-filvisning
 GenericName[ta]=Qt  வடிவமைப்பவர் UI கோப்புகள் காட்சியகம்
 GenericName[tg]=Намоишгари UI-файлҳо Qt Designer
diff -urN -x CVS kdesdk.orig/poxml/split.cpp kdesdk/poxml/split.cpp
--- kdesdk.orig/poxml/split.cpp	Fri Jan 24 09:28:03 2003
+++ kdesdk/poxml/split.cpp	Sun Oct 17 12:27:52 2004
@@ -111,8 +111,8 @@
 
         }
         cout << "\n";
-        outputMsg("msgid", (*it).msgid);
-        outputMsg("msgstr", (*it).msgstr);
+        outputMsg("msgid", StructureParser::descapeLiterals( (*it).msgid ));
+        outputMsg("msgstr", StructureParser::descapeLiterals( (*it).msgstr ));
         cout << "\n";
     }
 
diff -urN -x CVS kdesdk.orig/scripts/Makefile.am kdesdk/scripts/Makefile.am
--- kdesdk.orig/scripts/Makefile.am	Mon Jun 14 20:20:46 2004
+++ kdesdk/scripts/Makefile.am	Sun Oct  3 13:15:45 2004
@@ -9,7 +9,7 @@
   kdedoc qtdoc extractrc zonetab2pot.py \
   licensecheck fixkdeincludes fixuifiles includemocs create_cvsignore \
   cxxmetric extend_dmalloc kdekillall kdelnk2desktop.py \
-  package_crystalsvg png2mng.pl kdemangen.pl
+  package_crystalsvg png2mng.pl kdemangen.pl cvsforwardport
 
 # sorted by bin_SCRIPTS reference
 man_MANS = \
diff -urN -x CVS kdesdk.orig/scripts/README kdesdk/scripts/README
--- kdesdk.orig/scripts/README	Mon Jun 14 14:17:44 2004
+++ kdesdk/scripts/README	Sun Oct  3 13:15:45 2004
@@ -73,6 +73,8 @@
 
 cvsbackport	Backport the last commit in HEAD, to a branch.
 
+cvsforwardport  Forwardport the last commit in a branch to HEAD
+
 gendebug        Generate C++/Qt code for printing debug output.
 
 pruneemptydirs	Detects stale source dirs in a CVS tree
@@ -130,4 +132,4 @@
 Just add it to this README. For easy man pages see the perlpod manpage; the
 man page for many of these tools is in the script itself.
 
-$Id$
+$Id$
diff -urN -x CVS kdesdk.orig/scripts/cvsforwardport kdesdk/scripts/cvsforwardport
--- kdesdk.orig/scripts/cvsforwardport	Thu Jan  1 01:00:00 1970
+++ kdesdk/scripts/cvsforwardport	Sun Oct  3 13:15:45 2004
@@ -0,0 +1,32 @@
+#!/bin/sh
+# Forwardport the last change in a branch to HEAD
+# Usage: cvsforwardport <files>
+# WARNING: the branch tag is hardcoded into the script, make sure to check it!
+#
+# Initial author: Dirk Mueller
+# Support for multiple command-line arguments: David Faure
+
+BRANCH=KDE_3_3_BRANCH
+echo "Forwardporting to HEAD"
+TMPFILE=`mktemp cvsforwardport.XXXXXX` || exit 1
+files=$*
+until test $# -eq 0; do
+
+  echo "looking for last change to $1..."
+  cvslastchange $1 > $TMPFILE
+  echo "browsing last change to $1..."
+  less $TMPFILE
+  cvs up -A $1
+  patch < $TMPFILE
+  rm -f $TMPFILE
+  echo "showing diff for $1..."
+  cvs diff $1 | less
+
+  shift
+done
+
+echo "Press ENTER now to commit ($files) or Ctrl+C to abort"
+read confirm
+
+cvs commit $files
+cvs up -r $BRANCH $files
